<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.280">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="daiuc">
<meta name="dcterms.date" content="2023-01-03">

<title>NeuronReporters - CRISPR Screen RRA Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../site_libs/datatables-binding-0.24/datatables.js"></script>
<script src="../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">NeuronReporters</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#compute-enrichment-or-depletion" id="toc-compute-enrichment-or-depletion" class="nav-link" data-scroll-target="#compute-enrichment-or-depletion">Compute enrichment or depletion</a></li>
  <li><a href="#monte-carlo-simulation" id="toc-monte-carlo-simulation" class="nav-link" data-scroll-target="#monte-carlo-simulation">Monte Carlo simulation</a>
  <ul class="collapse">
  <li><a href="#enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown" id="toc-enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown" class="nav-link" data-scroll-target="#enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown">Enrichment of Td- vs.&nbsp;iN at Day14, with MC computed FDR, genes with FDR &lt; 0.15 are shown</a></li>
  <li><a href="#enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr" id="toc-enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr" class="nav-link" data-scroll-target="#enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr">Enrichment of Td- vs.&nbsp;Td+ at Day14, with MC computed FDR</a></li>
  <li><a href="#depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr" id="toc-depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr" class="nav-link" data-scroll-target="#depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr">Depletion of Td+ vs.&nbsp;iN at Day14, with MC computed FDR</a></li>
  </ul></li>
  <li><a href="#compare-with-hits-found-from-existing-method-using-fdr-.15" id="toc-compare-with-hits-found-from-existing-method-using-fdr-.15" class="nav-link" data-scroll-target="#compare-with-hits-found-from-existing-method-using-fdr-.15">Compare with hits found from existing method, using FDR&lt; .15</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">CRISPR Screen RRA Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>daiuc </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 3, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p><strong>Td- vs.&nbsp;iN at Day14 (S8/S6):</strong></p>
<ul>
<li>These cells became non-neurons, the <strong>enrichment</strong> of guides found from S8 / S6 means that these guides discrupt TFs that may be crucial to neuron development. Because these cells failed to become neurons, <strong>enrichment</strong> of these guides mean TFs responsible for neuron development are disrupted.</li>
</ul>
<p><strong>Td- vs.&nbsp;Td+ at Day14 (S8/S7)</strong></p>
<ul>
<li>S8 are non-neurons and S7 are neurons. The <strong>enrichment</strong> of guides found in neurons, while depleted in non-neurons, mean that these guides disrupte the function of TFs that are responsible for neuron development.</li>
</ul>
<p><strong>Td+ vs.&nbsp;iN at Day14 (S7/S6)</strong></p>
<ul>
<li><p>These cells became neurons. The <strong>enrichment</strong> of guides found from S7 - S6 means that these guides disrupt TFs that may be repressors of neuron development, or that these targetted TFs are reponsible for maintaining stem cell status.</p></li>
<li><p>The <strong>depletion</strong> of these guides means that the targeted TFs may be crucial to neuron development (since these cells have become neurons, you wouldnâ€™t find many cells with neuron related TFs being disrupted, thus you observe depletion of gRNAs).</p></li>
</ul>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.file = "../../resources/crispr/counts/RC.csv"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>cnts.file <span class="ot">=</span> <span class="st">"../code/resources/crispr/counts/wrangled_raw_counts.tsv"</span> <span class="co"># raw file</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.rn.file = "../../resources/crispr/counts/wrangled_ranknorm_counts.tsv" # rank normalized</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.l2.file = "/gpfs/commons/groups/sanjana_lab/cdai/TFscreen/RCmNorRatioLog2.lu.csv"</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> <span class="fu">fread</span>(cnts.file) <span class="sc">%&gt;%</span> .[, .(<span class="at">ID=</span>sgRNA, Gene, <span class="at">S6 =</span> Day14_iN, <span class="at">S7=</span>Day14_Tdpos, <span class="at">S8=</span>Day14_Tdneg)]</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.l2 = fread(cnts.l2.file) %&gt;% .[, .(ID, Gene, S6, S7, S8)]</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># group 10 NTC guides into 1 NTC group</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>NTC <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"NonTargeting_Human_"</span>, <span class="fu">sort</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">10</span>)), <span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>cnts[<span class="fu">str_detect</span>(<span class="fu">toupper</span>(Gene), <span class="st">"NONTARGET"</span>), Gene <span class="sc">:</span><span class="er">=</span> NTC]</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any guides where S6 or iN is 1 or less</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes that have less than 5 guides left at iN or S6</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> cnts[S6 <span class="sc">&gt;</span> <span class="dv">1</span>]</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> cnts[, .(ID, S6, S7, S8, .N), by <span class="ot">=</span> Gene][N<span class="sc">&gt;</span><span class="dv">4</span>, <span class="sc">-</span><span class="fu">c</span>(<span class="st">"N"</span>)]</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co"># log2 RPM normalize counts</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>cnts.norm <span class="ot">=</span> cnts[, <span class="fu">c</span>(</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>), .SDcols <span class="ot">=</span> selcols]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>makeDT <span class="ot">=</span> <span class="cf">function</span>(dt, dt.norm) {</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dt: is raw counts,</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dt.norm: is the same, but normalized</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># columns: ID, Gene, denom, num</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># both denom and num conditions must have at least 2 guides with reads &gt; 0</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    genes1 <span class="ot">=</span> dt[denom <span class="sc">&gt;</span> <span class="dv">0</span>, .N, by <span class="ot">=</span> Gene][N<span class="sc">&gt;</span><span class="dv">1</span>, Gene] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    genes2 <span class="ot">=</span> dt[num <span class="sc">&gt;</span> <span class="dv">0</span>, .N, by <span class="ot">=</span> Gene][N<span class="sc">&gt;</span><span class="dv">1</span>, Gene] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">intersect</span>(genes1, genes2)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[Gene <span class="sc">%in%</span> genes <span class="sc">&amp;</span> denom <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> num <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    dt.norm <span class="ot">=</span> dt.norm[Gene <span class="sc">%in%</span> genes]</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    dt.out <span class="ot">=</span> dt.norm[dt[, .(ID, Gene)], on <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"ID"</span>), nomatch<span class="ot">=</span><span class="cn">NULL</span>]</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(dt.out)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>weightedSum <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># x must be a numerical vector of exactly 2 elements</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># x must be sorted, x[1] &lt;= x[2]</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fl">0.25</span><span class="sc">*</span>x[<span class="dv">1</span>] <span class="sc">+</span> <span class="fl">0.75</span><span class="sc">*</span>x[<span class="dv">2</span>]) <span class="co"># more weights no the second rank</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>RRA <span class="ot">=</span> <span class="cf">function</span>(dt, enrichment) {</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dt is a dataframe with column ID, Gene, denom, num (which are Log2 RPM normalized reads)</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[, .(ID, Gene, denom, num, <span class="at">l2fc =</span> num <span class="sc">-</span> denom)] <span class="co"># compute l2fc</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (enrichment) {</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>        dt <span class="ot">=</span> dt[, .(ID, Gene,l2fc, <span class="at">rk=</span><span class="fu">rank</span>(<span class="sc">-</span>l2fc, <span class="at">ties.method =</span> <span class="st">"max"</span>))] <span class="co"># look for enriched guides</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>        dt <span class="ot">=</span> dt[, .(ID, Gene,l2fc, <span class="at">rk=</span><span class="fu">rank</span>(l2fc, <span class="at">ties.method =</span> <span class="st">"max"</span>))] <span class="co"># look for depleted guides</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[<span class="fu">order</span>(Gene, rk), .SD[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], by <span class="ot">=</span> Gene <span class="co"># select top 2 guides for each gene</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>            ][, .(<span class="at">rankscore =</span> <span class="fu">weightedSum</span>(rk)), by <span class="ot">=</span> Gene</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>             ][<span class="fu">order</span>(rankscore)]</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(dt)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<hr>
</section>
<section id="compute-enrichment-or-depletion" class="level2">
<h2 class="anchored" data-anchor-id="compute-enrichment-or-depletion">Compute enrichment or depletion</h2>
<p>Result shown in the next section.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enrichment of guides in td- vs. iN (S8 VS. S6)</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>dtS86 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)],</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)])</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>rraS86 <span class="ot">=</span> <span class="fu">RRA</span>(dtS86, <span class="at">enrichment =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enrichment of guides in td- vs. td+ (S8 VS. S7)</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>dtS87 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> cnts[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)],</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)])</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>rraS87 <span class="ot">=</span> <span class="fu">RRA</span>(dtS87, <span class="at">enrichment =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># depletion of guides in td+ vs. iN (S7 VS. S6)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>dtS76 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)],</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)])</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>rraS76 <span class="ot">=</span> <span class="fu">RRA</span>(dtS76, <span class="at">enrichment =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<hr>
</section>
<section id="monte-carlo-simulation" class="level2">
<h2 class="anchored" data-anchor-id="monte-carlo-simulation">Monte Carlo simulation</h2>
<p>Simulations are set to run 20 iterations.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>N_sim <span class="ot">=</span> <span class="dv">20</span> <span class="co"># number of simulations</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>sim.rraS86.l <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N_sim) {</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    sim.cnts <span class="ot">=</span> cnts[</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        , .(Gene, ID,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">S6 =</span> S6[<span class="fu">sample</span>(<span class="fu">length</span>(S6))],</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">S7 =</span> S7[<span class="fu">sample</span>(<span class="fu">length</span>(S7))],</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">S8 =</span> S8[<span class="fu">sample</span>(<span class="fu">length</span>(S8))])]</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># log2 RPM normalize counts</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    sim.cnts.norm <span class="ot">=</span> sim.cnts[, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )), </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>        .SDcols <span class="ot">=</span> selcols]</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    sim.dtS86 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> sim.cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)],</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> sim.cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)])</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    sim.rraS86 <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">RRA</span>(sim.dtS86, <span class="at">enrichment =</span> T))</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    sim.rraS86.l <span class="ot">=</span> <span class="fu">c</span>(sim.rraS86.l, sim.rraS86)</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co"># get ecdf</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>ecdf.S86 <span class="ot">=</span> <span class="fu">ecdf</span>(<span class="fu">map</span>(sim.rraS86.l, <span class="sc">~</span>.x<span class="sc">$</span>rankscore) <span class="sc">%&gt;%</span> flatten <span class="sc">%&gt;%</span> unlist)</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co"># use ecdf to compute FDR value</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>rraS86[, FDR <span class="sc">:</span><span class="er">=</span> <span class="fu">ecdf.S86</span>(rankscore)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>N_sim <span class="ot">=</span> <span class="dv">20</span> <span class="co"># number of simulations</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>sim.rraS87.l <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N_sim) {</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    sim.cnts <span class="ot">=</span> cnts[</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>        , .(Gene, ID,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">S6 =</span> S6[<span class="fu">sample</span>(<span class="fu">length</span>(S6))],</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">S7 =</span> S7[<span class="fu">sample</span>(<span class="fu">length</span>(S7))],</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">S8 =</span> S8[<span class="fu">sample</span>(<span class="fu">length</span>(S8))])]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># log2 RPM normalize counts</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    sim.cnts.norm <span class="ot">=</span> sim.cnts[, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )), </span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>        .SDcols <span class="ot">=</span> selcols]</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    sim.dtS87 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> sim.cnts[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)],</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> sim.cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)])</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    sim.rraS87 <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">RRA</span>(sim.dtS87, <span class="at">enrichment =</span> T))</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    sim.rraS87.l <span class="ot">=</span> <span class="fu">c</span>(sim.rraS87.l, sim.rraS87)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co"># get ecdf</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>ecdf.S87 <span class="ot">=</span> <span class="fu">ecdf</span>(<span class="fu">map</span>(sim.rraS87.l, <span class="sc">~</span>.x<span class="sc">$</span>rankscore) <span class="sc">%&gt;%</span> flatten <span class="sc">%&gt;%</span> unlist)</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co"># use ecdf to compute FDR value</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>rraS87[, FDR <span class="sc">:</span><span class="er">=</span> <span class="fu">ecdf.S87</span>(rankscore)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>N_sim <span class="ot">=</span> <span class="dv">20</span> <span class="co"># number of simulations</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>sim.rraS76.l <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N_sim) {</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    sim.cnts <span class="ot">=</span> cnts[</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>        , .(Gene, ID,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">S6 =</span> S6[<span class="fu">sample</span>(<span class="fu">length</span>(S6))],</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">S7 =</span> S7[<span class="fu">sample</span>(<span class="fu">length</span>(S7))],</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">S8 =</span> S8[<span class="fu">sample</span>(<span class="fu">length</span>(S8))])]</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># log2 RPM normalize counts</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    sim.cnts.norm <span class="ot">=</span> sim.cnts[, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )), </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>        .SDcols <span class="ot">=</span> selcols]</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    sim.dtS76 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> sim.cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)],</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> sim.cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)])</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    sim.rraS76 <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">RRA</span>(sim.dtS76, <span class="at">enrichment =</span> F))</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    sim.rraS76.l <span class="ot">=</span> <span class="fu">c</span>(sim.rraS76.l, sim.rraS76)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co"># get ecdf</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>ecdf.S76 <span class="ot">=</span> <span class="fu">ecdf</span>(<span class="fu">map</span>(sim.rraS76.l, <span class="sc">~</span>.x<span class="sc">$</span>rankscore) <span class="sc">%&gt;%</span> flatten <span class="sc">%&gt;%</span> unlist)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="co"># use ecdf to compute FDR value</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>rraS76[, FDR <span class="sc">:</span><span class="er">=</span> <span class="fu">ecdf.S76</span>(rankscore)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown" class="level3">
<h3 class="anchored" data-anchor-id="enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown">Enrichment of Td- vs.&nbsp;iN at Day14, with MC computed FDR, genes with FDR &lt; 0.15 are shown</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>rraS86[<span class="sc">!</span><span class="fu">is.na</span>(rankscore) <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> .<span class="dv">15</span>] <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>() <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">formatRound</span>(<span class="st">"FDR"</span>, <span class="at">digits =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="htmlwidget-5ca5bda3d45657cf287e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ca5bda3d45657cf287e">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102"],["ZNF441","ZNF440","ZBTB18","CRY1","BRWD1","CDX4","ZZZ3","E2F1","RAI14","EGR3","CHD5","ZNF250","PRDM6","SCRT2","TOX4","ZNF639","TAF1L","ZNF311","DLX5","ZIC1","BHLHA15","ZNF671","HSFY1","CBX4","THRA","VAX2","UBE2K","KLF10","EMX1","EBF1","JUNB","NR1I2","POLR2B","CREBZF","ZNF141","ARID1B","NonTargeting_Human_67","NKX3-2","RBPJ","ZNF510","NonTargeting_Human_31","CTDP1","ZNF554","INSM1","VGLL2","GATA1","HTATIP2","ISL2","TFEB","TSHZ3","JUND","MEOX2","ZNF136","TSC22D3","NonTargeting_Human_99","PRDM5","ZFPM1","ZNF239","NonTargeting_Human_22","PHF12","NFKBIB","MECP2","APBB2","SIX5","HIST1H2BC","ZNF766","HIF3A","PAX7","NEUROG3","ELK3","PAX2","LHX4","NonTargeting_Human_90","CTNNB1","NonTargeting_Human_13","SMAD3","ZKSCAN3","TAF5","RORB","ZNF502","SKI","FOXQ1","HOXC13","ZIM3","HOXB3","NonTargeting_Human_64","POU2F3","POLR2J","NFAT5","TBPL1","AIP","NonTargeting_Human_88","ZNF282","ZNF92","AHR","CNOT6","ZFY","ZNF34","ZNF547","RCOR2","NonTargeting_Human_57","NonTargeting_Human_73"],[28.75,49.25,83.75,88,88.25,89.75,98.5,102.75,115.5,123.25,123.75,127.75,132.75,142,142,144,149.5,149.75,154.25,155.25,163.75,165.75,167.25,170.5,171.5,174,175,179.5,180,182.5,186.25,198.75,209.25,212.25,213.75,215.25,219,219.75,219.75,233,233.25,237,239.75,247.25,247.25,251.25,254,254.5,257,259,271,271,273,274.25,274.75,284.5,284.5,285,285.5,288.25,294.25,298,298.5,302.5,303.25,308.5,309,310.5,315,317,326.5,327.5,330.25,334.75,335,337.75,338.75,339.5,343.25,348,349.25,357.75,361,362.25,363.5,368.75,370.5,377.25,378.5,379.5,389.75,393.75,395.75,396.25,405.5,408,409,409,410,411.25,412.5,417.25],[0.000843364256689412,0.00191673694702139,0.00621022770834931,0.00690025300927701,0.00690025300927701,0.00697692248715786,0.00897032891206011,0.0102737100360347,0.0124971248945795,0.014873878708886,0.0151038871425286,0.0157939124434563,0.0167906156559074,0.0189373610365713,0.0189373610365713,0.0195507168596182,0.0207774285057119,0.0209307674614736,0.022310818063329,0.022310818063329,0.0246109023997547,0.0254542666564441,0.0258376140458483,0.027064325691942,0.0271409951698229,0.0277543509928697,0.0279076899486315,0.0296710879398911,0.0298244268956528,0.0312044774975082,0.0325845280993636,0.0365713409491681,0.0397914590201641,0.0405581537989726,0.0412481790999003,0.0415548570114238,0.0427815686575174,0.0428582381353983,0.0428582381353983,0.0473050678524879,0.0475350762861305,0.0492218047995093,0.0505251859234839,0.0535919650387181,0.0535919650387181,0.0555853714636203,0.0568887525875949,0.0571954304991183,0.0581921337115694,0.0593421758797823,0.0639423445526336,0.0639423445526336,0.0650923867208464,0.0659357509775358,0.0662424288890593,0.071149275473434,0.071149275473434,0.0714559533849574,0.0716092923407192,0.0729126734646937,0.0764394694472131,0.078049528482711,0.0783562063942344,0.0801196043854941,0.0805796212527793,0.0837997393237752,0.0839530782795369,0.0847197730583455,0.087096526872652,0.0888599248639117,0.0932300851031204,0.0937667714482864,0.0955301694395461,0.0977535842980909,0.0979069232538526,0.0988269569884229,0.0993636433335889,0.0999769991566357,0.102047075059419,0.104500498351606,0.105420532086176,0.111017403971479,0.112704132484858,0.113470827263666,0.11469753890976,0.117150962201947,0.11830100437016,0.122364486697846,0.123361189910297,0.124127884689105,0.130491451353216,0.132868205167523,0.134248255769378,0.134631603158782,0.141685195123821,0.143525262592962,0.14429195737177,0.14429195737177,0.145211991106341,0.146285363796673,0.146822050141839,0.149888829257073]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>rankscore<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 5, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr" class="level3">
<h3 class="anchored" data-anchor-id="enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr">Enrichment of Td- vs.&nbsp;Td+ at Day14, with MC computed FDR</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>rraS87[<span class="sc">!</span><span class="fu">is.na</span>(rankscore) <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> .<span class="dv">15</span>] <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>() <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">formatRound</span>(<span class="st">"FDR"</span>, <span class="at">digits =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="htmlwidget-2e37e77da661efac278a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2e37e77da661efac278a">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94"],["ZNF441","ZNF440","CRY1","E2F1","CDX4","BHLHA15","ZNF311","EGR3","ZIC1","NR1I2","ZNF250","ZNF639","PRDM6","DLX5","TOX4","THRA","BRWD1","CHD5","CBX4","VAX2","ZBTB18","RAI14","RBPJ","TSHZ3","NKX3-2","TAF1L","APBB2","SCRT2","HSFY1","ZNF136","EMX1","UBE2K","CTDP1","JUNB","MECP2","POLR2B","ZZZ3","NonTargeting_Human_22","ZNF554","KLF10","INSM1","MEOX2","ZNF671","NonTargeting_Human_90","SMAD3","EBF1","ZNF563","PAX7","ZNF141","ZFPM1","HIST1H2BC","JUND","HTATIP2","PRDM5","GATA1","TFEB","CREBZF","VGLL2","SKI","ZFY","HIF3A","NonTargeting_Human_13","ELK3","NonTargeting_Human_31","NonTargeting_Human_57","ZIM3","ARID1B","AHR","NonTargeting_Human_88","NFAT5","NEUROG3","TSC22D3","LHX4","NonTargeting_Human_69","ZNF239","NonTargeting_Human_39","SIX5","ZNF510","PAX2","ZNF92","CTNNB1","TAF5","NFKBIB","NonTargeting_Human_99","ZKSCAN3","AFF4","ATF6B","ARNTL","TCEA2","PHF12","MPHOSPH8","ZNF34","RORB","HOXB3"],[10.75,53.25,54.25,77.75,81.25,84.5,95.5,105.5,107.75,112.75,133.25,133.25,135,136.75,139.75,140,143.5,145,162.5,162.5,164.75,165,166.25,171,173.75,176.75,178.5,180,183.25,190,197.75,197.75,204.25,206.25,209.25,210,210,212.5,214,220.5,229.5,232,232,236.5,239,240,250.25,252,253,258.75,264,267.75,268.75,280.25,295,296.5,297.5,297.75,298,298,301.75,306,318,319,319.25,320.25,323.5,324.75,325.25,328.5,333.25,335.5,341,341.75,343.25,346.75,347.25,355.25,357,362.5,363.75,365.25,368,369.75,380,380.75,381,381.25,382.25,384.5,385.25,390.75,397.25,399.25],[0.000164284540824708,0.00312140627566946,0.00328569081649417,0.00599638574010186,0.00657138163298834,0.00739280433711188,0.00952850336783309,0.0113356333169049,0.0116642023985543,0.0124856251026778,0.0184820108427797,0.0184820108427797,0.0186462953836044,0.0192212912764909,0.0197962871693774,0.0197962871693774,0.0209462789551503,0.0213569903072121,0.0264498110727781,0.0264498110727781,0.0273533760473139,0.0274355183177263,0.0279283719402004,0.0295712173484475,0.030146213241334,0.0312140627566946,0.0317890586495811,0.0321997700016428,0.0330211927057664,0.0358140298997864,0.0385247248233941,0.0385247248233941,0.0404961393132906,0.0417282733694759,0.0428782651552489,0.0435354033185477,0.0435354033185477,0.0441103992114342,0.0453425332676195,0.0483817972728766,0.0517496303597831,0.0531460489567932,0.0531460489567932,0.0547888943650403,0.0557746016099885,0.0561031706916379,0.060292426482668,0.0611138491867915,0.0613602759980286,0.0645638245441104,0.0669459503860687,0.0692459339576146,0.0696566453096764,0.0754887465089535,0.0832922621981272,0.083949400361426,0.0841958271726631,0.0844422539839001,0.0846065385247248,0.0846065385247248,0.0872350911779202,0.0894529324790537,0.0964350254641038,0.0966814522753409,0.0968457368161656,0.0975028749794644,0.0988171513060621,0.0992278626581239,0.0993921471989486,0.101774273040907,0.104567110234927,0.105470675209463,0.108263512403483,0.108838508296369,0.109495646459668,0.11130277640874,0.111631345490389,0.116477739444718,0.117545588960079,0.120666995235748,0.121570560210284,0.122966978807294,0.124527681945129,0.12559553146049,0.133234762608839,0.133727616231313,0.134302612124199,0.134466896665024,0.135288319369147,0.13676688023657,0.137177591588632,0.141531131920486,0.146706094956465,0.147691802201413]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>rankscore<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 5, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr" class="level3">
<h3 class="anchored" data-anchor-id="depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr">Depletion of Td+ vs.&nbsp;iN at Day14, with MC computed FDR</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>rraS76[FDR <span class="sc">&lt;</span> .<span class="dv">15</span>] <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>() <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">formatRound</span>(<span class="st">"FDR"</span>, <span class="at">digits =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="htmlwidget-af087bde037b30ad06aa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-af087bde037b30ad06aa">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310"],["MXI1","NCOA3","NR5A2","TFEC","HIF1AN","ZNF85","MCM7","PRDM5","SMARCA5","ZFPL1","ZNF582","ZNF337","HDAC9","ANKZF1","MAML3","RBBP7","NKX2-1","EIF2AK2","MYBBP1A","EWSR1","MCM3","NOTCH3","NUPR1","ZNF107","SRF","GFI1","ARID1B","STAT4","YEATS2","ZNF207","GRHL1","EGR1","HNRNPUL1","MNX1","TBX6","PHF6","PHF7","LZTR1","TAF6","SMARCB1","GTF2B","ZNF462","BCL3","JMJD1C","MED21","KMT2D","ZNF256","ZNF469","NCOA1","BCL11A","PIR","ATF6B","ZNF100","IRX4","FOXD3","PAX8","ID3","PRDM9","ZNF235","ZNF383","ANKRD42","PRDM13","ZNF292","NCOA2","ZSCAN18","TRIP11","TIAL1","MZF1","ZNF580","HNRNPD","NPAS3","NDNL2","IKBKG","RBL1","BARX1","KDM2A","TOX2","GMEB2","SNRPB","RERE","TCF21","CERS2","ZBTB10","ZNF80","BAZ1B","MED1","HIC2","NFIB","ZNF573","PRDM14","HES6","LDB1","TBX18","HMGB3","UBP1","BARX2","ISL2","ZRANB2","KDM4A","POLR3E","CUX1","NR4A2","HCLS1","BCOR","ZDHHC16","ZSWIM4","DAXX","ARNT","SNAPC3","ZDHHC9","ZNF653","NFXL1","ATF1","CXXC1","TP53BP1","NELFE","YEATS4","INSM1","ZMYND11","MYCL","ZNF451","ZFP37","ZBTB14","SOX6","SOX2","SREBF2","MTA2","MTF2","ZZZ3","TMF1","COPS5","PHTF1","TBX22","TCEA2","NFKBIZ","NKD2","MAFF","MXD1","DAZAP2","ARNTL2","NELFB","URI1","FOXN2","SFSWAP","MED8","SUGP1","SCRT2","BARHL2","CREBZF","TFDP2","NKRF","EBF3","PHF21A","TLE6","RAX2","BNC1","CDX2","GTF2E1","SCML2","CSRP3","CDX1","RBBP4","GATAD2B","ZNF500","PTMA","ARID3B","SP100","MSX2","ZNF324","HIST1H1T","TONSL","ASCC1","TAL2","FOXP3","ZNF592","PDLIM4","TBX15","HINFP","TLX1","SMARCC1","ZNF221","SPI1","SIX5","GATAD1","EGR2","FIGLA","CEBPE","ZNF638","MPLKIP","SOWAHC","TRIM5","ATF7IP","CTNNBL1","ASCL3","MCM5","MED15","POLR3F","EAF1","ARNT2","HOXB5","ZNF593","PHF23","MYEF2","SRSF5","ZNF641","MCM8","NFATC2","NR0B1","ADNP2","YAF2","TCEB3C","ZKSCAN7","HOXC5","ZNF446","TOP2B","TP63","UBTF","SIRT4","HMGN4","POU3F2","MYT1L","POLR1C","POU4F1","IRF5","ZNF32","MCM4","CEBPD","HSBP1","RFC1","MED23","MED27","ZNF490","ZSCAN23","ZBTB20","SNAI1","NR1H3","HAND1","EHMT1","TRIM31","MYB","FOXK2","POU6F1","FOXN4","SMAD6","WT1","CTCF","IRF3","MITF","ANKRA2","TFAP4","PAX4","ZSCAN10","TSHZ2","SUPT5H","ENO1","VSX1","ZNF343","PMS1","ABT1","ZNF34","RARB","CALCOCO1","SAP30BP","ZNF700","ZNF480","ZBTB3","LEF1","DEK","NonTargeting_Human_57","ZNF566","GATA3","TAF4","RFX3","KMT2A","FOSL2","ASB8","GLIS1","TRIM25","HIRA","HEYL","VAX1","TAF9B","CNBP","BLZF1","IRF7","ZNF143","CASZ1","GTF3C3","TBX1","IRX2","BRCA1","HIST1H2BJ","NR1D2","ZFPM1","NR1D1","EPC1","NonTargeting_Human_12","ZNF584","BAZ1A","NEUROG1","ZFP57","ZNF7","HMG20A","HOXB8","EZH2","NFKB1","TBX5","ANKRD1","ZNF229","SP4"],[62.5,65,70.25,84,87.5,88.5,91,122,142.5,145.5,147.5,164.75,174.75,186.25,186.25,207.75,209.75,214.25,215.75,219.25,220.75,230.75,233,241.75,243.5,253.25,255.75,256.25,257,260,261,273.75,278.5,301.75,305.75,306.75,309.5,312,313.5,324.25,332.75,338.5,341,344.75,352,362.75,364.75,366.5,367.25,370.75,371.75,375,378,382,394.25,396,407,407.75,409.75,412,419.5,419.75,426.25,432,434.25,434.75,436,439.5,439.5,440.25,440.25,441.25,441.5,443,446.75,450,455.75,456.5,456.75,457,457.5,458.75,460.25,468.25,474.25,475,477,483.75,487.5,488.5,490.5,501.75,504.75,510.25,515.25,529,529,535.75,554,558.5,560.75,561.75,567.75,571.5,572.25,578,587.5,591.5,600.75,601.75,602.25,603.5,607.25,609.75,619.75,620.5,620.75,623.75,625.25,632,632.25,639.5,640.75,641.75,642,642.75,645.25,652,654.5,658.75,660,660.75,663.75,674,681,681.25,693.25,694.75,695.25,701,703.75,707.75,718.25,718.25,724.75,725.75,728.75,729.75,735,737,739.75,741,743,746.25,746.75,747.25,754.75,756.75,770.75,780,786.5,799.75,806,812.25,822.5,825.5,829.25,834,835.75,838.25,843.25,843.5,845.75,850,854.75,855.75,861,867.5,868.75,869,873.75,877.75,885.75,886,887.5,892.75,895.25,896,901.75,903,907.75,909,909.75,920,920.5,921.75,923,928,930.75,931,937,943,952.75,954,958,959.5,961.25,966.75,972.75,972.75,974.5,978.25,979.75,980.25,981,990.25,990.75,999,1010,1010.5,1016.5,1021.25,1024.5,1024.75,1028.25,1028.75,1029,1031.25,1033,1041,1043.75,1044.75,1047.75,1053,1054.75,1055.5,1056,1056.75,1056.75,1057.5,1057.75,1058.75,1059.5,1062,1064.5,1068.5,1068.75,1069,1070.25,1077,1077.25,1077.5,1078.25,1085.25,1091.75,1093.25,1093.5,1094,1096.5,1097.75,1098.25,1100,1101.5,1102.25,1103.25,1105,1105.75,1107,1112.75,1114.5,1115.75,1116.25,1116.75,1119.75,1121.25,1123.75,1124.75,1125,1125.5,1129.25,1131.25,1132.75,1133.5,1136.5,1138.25,1139.25,1144.25,1145.75,1147.25,1147.5,1150.75,1152.5,1154,1155.75,1157.25,1158,1159.25,1161,1163.75,1165.5,1168.25,1168.25,1175,1175.25,1176.25,1176.5,1176.5,1177.75,1178.25,1193.25],[0.000506329113924051,0.000531645569620253,0.000582278481012658,0.000911392405063291,0.0010126582278481,0.00113924050632911,0.00121518987341772,0.00217721518987342,0.00288607594936709,0.00313924050632911,0.00318987341772152,0.0040253164556962,0.00458227848101266,0.00506329113924051,0.00506329113924051,0.00620253164556962,0.00625316455696203,0.00650632911392405,0.00655696202531646,0.00670886075949367,0.00678481012658228,0.00741772151898734,0.00759493670886076,0.00827848101265823,0.00840506329113924,0.00934177215189873,0.00954430379746835,0.00956962025316456,0.00964556962025316,0.00982278481012658,0.00994936708860759,0.0108354430379747,0.0111392405063291,0.0129367088607595,0.0132658227848101,0.0133670886075949,0.0135949367088608,0.013873417721519,0.0141012658227848,0.0149620253164557,0.0156962025316456,0.0162784810126582,0.0164810126582278,0.0167088607594937,0.0172405063291139,0.018126582278481,0.018379746835443,0.0185569620253165,0.0186075949367089,0.0190126582278481,0.0191139240506329,0.0192658227848101,0.0195443037974684,0.0199493670886076,0.0211392405063291,0.0213670886075949,0.0224050632911392,0.0224050632911392,0.0226835443037975,0.0229113924050633,0.0237974683544304,0.0237974683544304,0.0244810126582278,0.0251645569620253,0.0253164556962025,0.0253417721518987,0.0255949367088608,0.0259240506329114,0.0259240506329114,0.0260253164556962,0.0260253164556962,0.0261012658227848,0.0262278481012658,0.0263544303797468,0.0266329113924051,0.0271139240506329,0.0275696202531646,0.027620253164557,0.0276455696202532,0.0277721518987342,0.0278227848101266,0.0278481012658228,0.0279493670886076,0.0291139240506329,0.0298481012658228,0.0299493670886076,0.0302784810126582,0.0311645569620253,0.0314430379746835,0.0316962025316456,0.0318227848101266,0.0331392405063291,0.0334177215189873,0.0340253164556962,0.0344556962025316,0.0361518987341772,0.0361518987341772,0.0370886075949367,0.039620253164557,0.0402278481012658,0.0405316455696203,0.0405822784810127,0.0412911392405063,0.041873417721519,0.042,0.0429873417721519,0.0440506329113924,0.0446329113924051,0.0457721518987342,0.0459746835443038,0.0460253164556962,0.0462025316455696,0.0468354430379747,0.0470126582278481,0.0482025316455696,0.048253164556962,0.0483291139240506,0.0488607594936709,0.0490632911392405,0.0499493670886076,0.05,0.0510379746835443,0.0511645569620253,0.0512658227848101,0.0512911392405063,0.0514430379746835,0.0518481012658228,0.0529113924050633,0.0532911392405063,0.0538227848101266,0.0538987341772152,0.0539746835443038,0.0544050632911392,0.055746835443038,0.0569620253164557,0.0569620253164557,0.0584050632911392,0.0586075949367089,0.0586582278481013,0.0594936708860759,0.059873417721519,0.0605569620253165,0.0621518987341772,0.0621518987341772,0.063493670886076,0.0636708860759494,0.0641772151898734,0.064379746835443,0.0649367088607595,0.0651645569620253,0.065620253164557,0.065873417721519,0.0662278481012658,0.0666075949367089,0.0666835443037975,0.0667088607594937,0.0680506329113924,0.0683291139240506,0.0703291139240506,0.0719746835443038,0.0729620253164557,0.0754177215189873,0.0764556962025316,0.0773164556962025,0.0791898734177215,0.0796708860759494,0.0805063291139241,0.081493670886076,0.081746835443038,0.0822784810126582,0.0831139240506329,0.0832405063291139,0.0836962025316456,0.0843291139240506,0.0850379746835443,0.0851645569620253,0.085873417721519,0.0868607594936709,0.0871139240506329,0.0871392405063291,0.087873417721519,0.0884810126582278,0.0898481012658228,0.0898481012658228,0.0900506329113924,0.0909113924050633,0.0911898734177215,0.0913417721518987,0.0922278481012658,0.0924303797468354,0.0933417721518987,0.0935696202531646,0.0937974683544304,0.0952911392405063,0.0953417721518987,0.0954430379746836,0.0956708860759494,0.0965569620253165,0.0970632911392405,0.0971139240506329,0.0983037974683544,0.099493670886076,0.101392405063291,0.101721518987342,0.102556962025316,0.102759493670886,0.102987341772152,0.103949367088608,0.104987341772152,0.104987341772152,0.105417721518987,0.106101265822785,0.106379746835443,0.106455696202532,0.106556962025316,0.108329113924051,0.108430379746835,0.11,0.111772151898734,0.111848101265823,0.112987341772152,0.113696202531646,0.114329113924051,0.114354430379747,0.114936708860759,0.115088607594937,0.115113924050633,0.115493670886076,0.115924050632911,0.117645569620253,0.118126582278481,0.118227848101266,0.118987341772152,0.12020253164557,0.12053164556962,0.120632911392405,0.120759493670886,0.120987341772152,0.120987341772152,0.121189873417722,0.121215189873418,0.121392405063291,0.121518987341772,0.121848101265823,0.122405063291139,0.123240506329114,0.123341772151899,0.123392405063291,0.123721518987342,0.125063291139241,0.125063291139241,0.125063291139241,0.125291139240506,0.12693670886076,0.128430379746835,0.128860759493671,0.129012658227848,0.129113924050633,0.129620253164557,0.129848101265823,0.129949367088608,0.130227848101266,0.130582278481013,0.130708860759494,0.130886075949367,0.13126582278481,0.13146835443038,0.131746835443038,0.133012658227848,0.133443037974684,0.133594936708861,0.133620253164557,0.133772151898734,0.134253164556962,0.134582278481013,0.135037974683544,0.135189873417722,0.135240506329114,0.135341772151899,0.135924050632911,0.136151898734177,0.136481012658228,0.136582278481013,0.13726582278481,0.137620253164557,0.137772151898734,0.13873417721519,0.139063291139241,0.139443037974684,0.139518987341772,0.140481012658228,0.140987341772152,0.141367088607595,0.141898734177215,0.14220253164557,0.142379746835443,0.142556962025316,0.142886075949367,0.143493670886076,0.14379746835443,0.144582278481013,0.144582278481013,0.145873417721519,0.145873417721519,0.146101265822785,0.146101265822785,0.146101265822785,0.146379746835443,0.146455696202532,0.149417721518987]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>rankscore<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 5, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
<hr>
</section>
</section>
<section id="compare-with-hits-found-from-existing-method-using-fdr-.15" class="level2">
<h2 class="anchored" data-anchor-id="compare-with-hits-found-from-existing-method-using-fdr-.15">Compare with hits found from existing method, using FDR&lt; .15</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>hitlist <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"../code/resources/crispr/Hitlist_20191230.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Previous method found 122 genes, shown below:</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-9b04d9509a66dd0a6b4f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9b04d9509a66dd0a6b4f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122"],["ZNF441","ZNF440","CRY1","E2F1","CDX4","BHLHA15","ZNF311","EGR3","ZIC1","NR1I2","ZNF639","ZNF250","PRDM6","DLX5","TOX4","THRA","BRWD1","CHD5","VAX2","CBX4","ZBTB18","RAI14","RBPJ","TSHZ3","NKX3-2","TAF1L","APBB2","SCRT2","HSFY1","ZNF136","EMX1","UBE2K","CTDP1","JUNB","MECP2","POLR2B","ZZZ3","ZNF554","KLF10","INSM1","ZNF671","MEOX2","SMAD3","EBF1","ZNF563","PAX7","ZNF141","ZFPM1","HIST1H2BC","JUND","HTATIP2","PRDM5","GATA1","TFEB","CREBZF","VGLL2","ZFY","SKI","HIF3A","ELK3","ZIM3","ARID1B","AHR","NFAT5","NEUROG3","TSC22D3","LHX4","ZNF239","SIX5","ZNF510","PAX2","ZNF92","CTNNB1","TAF5","NFKBIB","ZKSCAN3","AFF4","ARNTL","TCEA2","PHF12","MPHOSPH8","ZNF34","RORB","HOXB3","FOXQ1","POU2F3","ZNF502","ZNF324","SLA2","TBPL1","KLF14","ZBTB43","HOXC13","AIP","NBPF24","ZNF282","ISL2","CEBPA","PHOX2A","NKRF","SNAI2","FOXN2","MLLT6","RCOR2","NR1D2","ZNF547","POLR2J","SOX15","JADE3","ZFP2","POU3F2","ASB1","ZNF3","HIST1H1E","ZNF573","ZNF200","ZNF107","NEUROG1","CNOT6","GTF2A2","EHF","NEUROG2"],[10.75,53.25,54.25,77.75,81.25,84.5,95.5,105.5,107.75,112.75,133.25,133.25,135,136.75,139.75,140,143.5,145,162.5,162.5,164.75,165,166.25,171,173.75,176.75,178.5,180,183.25,190,197.75,197.75,204.25,206.25,209.25,210,210,214,220.5,229.5,232,232,239,240,250.25,252,253,258.75,264,267.75,268.75,280.25,295,296.5,297.5,297.75,298,298,301.75,318,320.25,323.5,324.75,328.5,333.25,335.5,341,343.25,347.25,355.25,357,362.5,363.75,365.25,368,380,380.75,381.25,382.25,384.5,385.25,390.75,397.25,399.25,414.25,421.25,426.5,428.25,429.75,432.75,433.25,435,435.25,438,440.5,450.75,451,453,457.75,461.75,469.25,470.5,471,474.25,474.75,479.5,483.75,487,495.25,511.25,526.75,528,529.75,529.75,540,544.25,564.5,802.25,857.75,null,null,null],[0.000164284540824708,0.00312140627566946,0.00328569081649417,0.00599638574010186,0.00657138163298834,0.00739280433711188,0.00952850336783309,0.0113356333169049,0.0116642023985543,0.0124856251026778,0.0184820108427797,0.0184820108427797,0.0186462953836044,0.0192212912764909,0.0197962871693774,0.0197962871693774,0.0209462789551503,0.0213569903072121,0.0264498110727781,0.0264498110727781,0.0273533760473139,0.0274355183177263,0.0279283719402004,0.0295712173484475,0.030146213241334,0.0312140627566946,0.0317890586495811,0.0321997700016428,0.0330211927057664,0.0358140298997864,0.0385247248233941,0.0385247248233941,0.0404961393132906,0.0417282733694759,0.0428782651552489,0.0435354033185477,0.0435354033185477,0.0453425332676195,0.0483817972728766,0.0517496303597831,0.0531460489567932,0.0531460489567932,0.0557746016099885,0.0561031706916379,0.060292426482668,0.0611138491867915,0.0613602759980286,0.0645638245441104,0.0669459503860687,0.0692459339576146,0.0696566453096764,0.0754887465089535,0.0832922621981272,0.083949400361426,0.0841958271726631,0.0844422539839001,0.0846065385247248,0.0846065385247248,0.0872350911779202,0.0964350254641038,0.0975028749794644,0.0988171513060621,0.0992278626581239,0.101774273040907,0.104567110234927,0.105470675209463,0.108263512403483,0.109495646459668,0.111631345490389,0.116477739444718,0.117545588960079,0.120666995235748,0.121570560210284,0.122966978807294,0.124527681945129,0.133234762608839,0.133727616231313,0.134466896665024,0.135288319369147,0.13676688023657,0.137177591588632,0.141531131920486,0.146706094956465,0.147691802201413,0.157384590110071,0.162723837686874,0.166995235748316,0.167898800722852,0.168966650238213,0.171266633809759,0.171923771973057,0.173238048299655,0.173320190570067,0.17479875143749,0.176359454575324,0.18317726301955,0.183423689830787,0.185477246591096,0.189584360111713,0.192459339576146,0.197223591260062,0.198291440775423,0.19853786758666,0.19993428618367,0.200427139806144,0.20305569245934,0.205355676030886,0.208230655495318,0.214062756694595,0.228601938557582,0.24108756366026,0.242401839986857,0.244044685395104,0.244044685395104,0.252751766058814,0.255544603252834,0.270905207819944,0.481107277805159,0.536306883522261,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>rankscore<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 5, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"venn.svg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="venn.svg" class="img-fluid"></p>
</div>
</div>
<p>Notice a large number of overlaps. When we combine hits found from Td- vs.&nbsp;iN and Td- vs.&nbsp;Td+ with FDR &lt; 0.15 (which is the same way we selected our 120 hits. There are 23 (NEUROG1 and NEUROG2 are not considered) hits found in existing method, but not found in the RRA method. Nearly all the hits that we initially removed in our existing method, which are labeled as hit with the RRA method are non-targeting controls. This highlights the advantage of our existing manual method of more precise control of false positives and potential false negatives. Of course, the disadvantage is that the existing manual method does not produce a p-value thus is less informative in statistical confidence.</p>
<p>Below shows hits that are found in both this method and our existing method. Notice many of these TFs are repeatedly identified as differentially expressed as well as later identified as nodes in the putative neuronal development gene regulatory network.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">intersect</span>(<span class="fu">union</span>(set87, set86), hitlist<span class="sc">$</span>gene_name) <span class="sc">%&gt;%</span> length</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 96</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">intersect</span>(<span class="fu">union</span>(set87, set86), hitlist<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ZNF441"    "ZNF440"    "CRY1"      "E2F1"      "CDX4"      "BHLHA15"  
 [7] "ZNF311"    "EGR3"      "ZIC1"      "NR1I2"     "ZNF250"    "ZNF639"   
[13] "PRDM6"     "DLX5"      "TOX4"      "THRA"      "BRWD1"     "CHD5"     
[19] "CBX4"      "VAX2"      "ZBTB18"    "RAI14"     "RBPJ"      "TSHZ3"    
[25] "NKX3-2"    "TAF1L"     "APBB2"     "SCRT2"     "HSFY1"     "ZNF136"   
[31] "EMX1"      "UBE2K"     "CTDP1"     "JUNB"      "MECP2"     "POLR2B"   
[37] "ZZZ3"      "ZNF554"    "KLF10"     "INSM1"     "MEOX2"     "ZNF671"   
[43] "SMAD3"     "EBF1"      "ZNF563"    "PAX7"      "ZNF141"    "ZFPM1"    
[49] "HIST1H2BC" "JUND"      "HTATIP2"   "PRDM5"     "GATA1"     "TFEB"     
[55] "CREBZF"    "VGLL2"     "SKI"       "ZFY"       "HIF3A"     "ELK3"     
[61] "ZIM3"      "ARID1B"    "AHR"       "NFAT5"     "NEUROG3"   "TSC22D3"  
[67] "LHX4"      "ZNF239"    "SIX5"      "ZNF510"    "PAX2"      "ZNF92"    
[73] "CTNNB1"    "TAF5"      "NFKBIB"    "ZKSCAN3"   "AFF4"      "ARNTL"    
[79] "TCEA2"     "PHF12"     "MPHOSPH8"  "ZNF34"     "RORB"      "HOXB3"    
[85] "ISL2"      "ZNF502"    "FOXQ1"     "HOXC13"    "POU2F3"    "POLR2J"   
[91] "TBPL1"     "AIP"       "ZNF282"    "CNOT6"     "ZNF547"    "RCOR2"    </code></pre>
</div>
</div>
<p>Below are hits that are found in our existing method but missed in this method.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(hitlist<span class="sc">$</span>gene_name, <span class="fu">union</span>(set87, set86))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "PHOX2A"   "SNAI2"    "ZBTB43"   "ZNF200"   "MLLT6"    "ZNF3"    
 [7] "ZNF324"   "ASB1"     "GTF2A2"   "JADE3"    "ZNF573"   "NBPF24"  
[13] "SOX15"    "FOXN2"    "CEBPA"    "NKRF"     "NR1D2"    "EHF"     
[19] "POU3F2"   "KLF14"    "SLA2"     "ZFP2"     "ZNF107"   "NEUROG1" 
[25] "NEUROG2"  "HIST1H1E"</code></pre>
</div>
</div>
<p>Below are labeled hits in this method but rejected in our existing method. Notice many are in fact non-targeting controls, suggesting our existing method may be better suited in removing false positives.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">union</span>(set87, set86), hitlist<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "NonTargeting_Human_22" "NonTargeting_Human_90" "NonTargeting_Human_13"
 [4] "NonTargeting_Human_31" "NonTargeting_Human_57" "NonTargeting_Human_88"
 [7] "NonTargeting_Human_69" "NonTargeting_Human_39" "NonTargeting_Human_99"
[10] "ATF6B"                 "NonTargeting_Human_67" "ZNF766"               
[13] "NonTargeting_Human_64" "NonTargeting_Human_73"</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>