<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.280">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="daiuc">
<meta name="dcterms.date" content="2023-01-03">

<title>NeuronReporters - CRISPR Screen RRA Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">NeuronReporters</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#compute-enrichment-or-depletion" id="toc-compute-enrichment-or-depletion" class="nav-link" data-scroll-target="#compute-enrichment-or-depletion">Compute enrichment or depletion</a></li>
  <li><a href="#monte-carlo-simulation" id="toc-monte-carlo-simulation" class="nav-link" data-scroll-target="#monte-carlo-simulation">Monte Carlo simulation</a>
  <ul class="collapse">
  <li><a href="#enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown" id="toc-enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown" class="nav-link" data-scroll-target="#enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown">Enrichment of Td- vs.&nbsp;iN at Day14, with MC computed FDR, genes with FDR &lt; 0.15 are shown</a></li>
  <li><a href="#enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr" id="toc-enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr" class="nav-link" data-scroll-target="#enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr">Enrichment of Td- vs.&nbsp;Td+ at Day14, with MC computed FDR</a></li>
  <li><a href="#depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr" id="toc-depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr" class="nav-link" data-scroll-target="#depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr">Depletion of Td+ vs.&nbsp;iN at Day14, with MC computed FDR</a></li>
  </ul></li>
  <li><a href="#compare-with-hits-found-from-existing-method-using-fdr-.15" id="toc-compare-with-hits-found-from-existing-method-using-fdr-.15" class="nav-link" data-scroll-target="#compare-with-hits-found-from-existing-method-using-fdr-.15">Compare with hits found from existing method, using FDR&lt; .15</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">CRISPR Screen RRA Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>daiuc </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 3, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p><strong>Td- vs.&nbsp;iN at Day14 (S8/S6):</strong></p>
<ul>
<li>These cells became non-neurons, the <strong>enrichment</strong> of guides found from S8 / S6 means that these guides discrupt TFs that may be crucial to neuron development. Because these cells failed to become neurons, <strong>enrichment</strong> of these guides mean TFs responsible for neuron development are disrupted.</li>
</ul>
<p><strong>Td- vs.&nbsp;Td+ at Day14 (S8/S7)</strong></p>
<ul>
<li>S8 are non-neurons and S7 are neurons. The <strong>enrichment</strong> of guides found in neurons, while depleted in non-neurons, mean that these guides disrupte the function of TFs that are responsible for neuron development.</li>
</ul>
<p><strong>Td+ vs.&nbsp;iN at Day14 (S7/S6)</strong></p>
<ul>
<li><p>These cells became neurons. The <strong>enrichment</strong> of guides found from S7 - S6 means that these guides disrupt TFs that may be repressors of neuron development, or that these targetted TFs are reponsible for maintaining stem cell status.</p></li>
<li><p>The <strong>depletion</strong> of these guides means that the targeted TFs may be crucial to neuron development (since these cells have become neurons, you wouldn’t find many cells with neuron related TFs being disrupted, thus you observe depletion of gRNAs).</p></li>
</ul>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.file = "../../resources/crispr/counts/RC.csv"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>cnts.file <span class="ot">=</span> <span class="st">"../code/resources/crispr/counts/wrangled_raw_counts.tsv"</span> <span class="co"># raw file</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.rn.file = "../../resources/crispr/counts/wrangled_ranknorm_counts.tsv" # rank normalized</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.l2.file = "/gpfs/commons/groups/sanjana_lab/cdai/TFscreen/RCmNorRatioLog2.lu.csv"</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> <span class="fu">fread</span>(cnts.file) <span class="sc">%&gt;%</span> .[, .(<span class="at">ID=</span>sgRNA, Gene, <span class="at">S6 =</span> Day14_iN, <span class="at">S7=</span>Day14_Tdpos, <span class="at">S8=</span>Day14_Tdneg)]</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cnts.l2 = fread(cnts.l2.file) %&gt;% .[, .(ID, Gene, S6, S7, S8)]</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># group 10 NTC guides into 1 NTC group</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>NTC <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"NonTargeting_Human_"</span>, <span class="fu">sort</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">10</span>)), <span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>cnts[<span class="fu">str_detect</span>(<span class="fu">toupper</span>(Gene), <span class="st">"NONTARGET"</span>), Gene <span class="sc">:</span><span class="er">=</span> NTC]</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any guides where S6 or iN is 1 or less</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes that have less than 5 guides left at iN or S6</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> cnts[S6 <span class="sc">&gt;</span> <span class="dv">1</span>]</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> cnts[, .(ID, S6, S7, S8, .N), by <span class="ot">=</span> Gene][N<span class="sc">&gt;</span><span class="dv">4</span>, <span class="sc">-</span><span class="fu">c</span>(<span class="st">"N"</span>)]</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co"># log2 RPM normalize counts</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>cnts.norm <span class="ot">=</span> cnts[, <span class="fu">c</span>(</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>), .SDcols <span class="ot">=</span> selcols]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>makeDT <span class="ot">=</span> <span class="cf">function</span>(dt, dt.norm) {</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dt: is raw counts,</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dt.norm: is the same, but normalized</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># columns: ID, Gene, denom, num</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># both denom and num conditions must have at least 2 guides with reads &gt; 0</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    genes1 <span class="ot">=</span> dt[denom <span class="sc">&gt;</span> <span class="dv">0</span>, .N, by <span class="ot">=</span> Gene][N<span class="sc">&gt;</span><span class="dv">1</span>, Gene] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    genes2 <span class="ot">=</span> dt[num <span class="sc">&gt;</span> <span class="dv">0</span>, .N, by <span class="ot">=</span> Gene][N<span class="sc">&gt;</span><span class="dv">1</span>, Gene] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">intersect</span>(genes1, genes2)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[Gene <span class="sc">%in%</span> genes <span class="sc">&amp;</span> denom <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> num <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    dt.norm <span class="ot">=</span> dt.norm[Gene <span class="sc">%in%</span> genes]</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    dt.out <span class="ot">=</span> dt.norm[dt[, .(ID, Gene)], on <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"ID"</span>), nomatch<span class="ot">=</span><span class="cn">NULL</span>]</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(dt.out)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>weightedSum <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># x must be a numerical vector of exactly 2 elements</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># x must be sorted, x[1] &lt;= x[2]</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fl">0.25</span><span class="sc">*</span>x[<span class="dv">1</span>] <span class="sc">+</span> <span class="fl">0.75</span><span class="sc">*</span>x[<span class="dv">2</span>]) <span class="co"># more weights no the second rank</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>RRA <span class="ot">=</span> <span class="cf">function</span>(dt, enrichment) {</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dt is a dataframe with column ID, Gene, denom, num (which are Log2 RPM normalized reads)</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[, .(ID, Gene, denom, num, <span class="at">l2fc =</span> num <span class="sc">-</span> denom)] <span class="co"># compute l2fc</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (enrichment) {</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>        dt <span class="ot">=</span> dt[, .(ID, Gene,l2fc, <span class="at">rk=</span><span class="fu">rank</span>(<span class="sc">-</span>l2fc, <span class="at">ties.method =</span> <span class="st">"max"</span>))] <span class="co"># look for enriched guides</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>        dt <span class="ot">=</span> dt[, .(ID, Gene,l2fc, <span class="at">rk=</span><span class="fu">rank</span>(l2fc, <span class="at">ties.method =</span> <span class="st">"max"</span>))] <span class="co"># look for depleted guides</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[<span class="fu">order</span>(Gene, rk), .SD[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], by <span class="ot">=</span> Gene <span class="co"># select top 2 guides for each gene</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>            ][, .(<span class="at">rankscore =</span> <span class="fu">weightedSum</span>(rk)), by <span class="ot">=</span> Gene</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>             ][<span class="fu">order</span>(rankscore)]</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(dt)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<hr>
</section>
<section id="compute-enrichment-or-depletion" class="level2">
<h2 class="anchored" data-anchor-id="compute-enrichment-or-depletion">Compute enrichment or depletion</h2>
<p>Result shown in the next section.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enrichment of guides in td- vs. iN (S8 VS. S6)</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>dtS86 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)],</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)])</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>rraS86 <span class="ot">=</span> <span class="fu">RRA</span>(dtS86, <span class="at">enrichment =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enrichment of guides in td- vs. td+ (S8 VS. S7)</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>dtS87 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> cnts[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)],</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)])</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>rraS87 <span class="ot">=</span> <span class="fu">RRA</span>(dtS87, <span class="at">enrichment =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># depletion of guides in td+ vs. iN (S7 VS. S6)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>dtS76 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)],</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)])</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>rraS76 <span class="ot">=</span> <span class="fu">RRA</span>(dtS76, <span class="at">enrichment =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<hr>
</section>
<section id="monte-carlo-simulation" class="level2">
<h2 class="anchored" data-anchor-id="monte-carlo-simulation">Monte Carlo simulation</h2>
<p>Simulations are set to run 20 iterations.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>N_sim <span class="ot">=</span> <span class="dv">20</span> <span class="co"># number of simulations</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>sim.rraS86.l <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N_sim) {</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    sim.cnts <span class="ot">=</span> cnts[</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        , .(Gene, ID,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">S6 =</span> S6[<span class="fu">sample</span>(<span class="fu">length</span>(S6))],</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">S7 =</span> S7[<span class="fu">sample</span>(<span class="fu">length</span>(S7))],</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">S8 =</span> S8[<span class="fu">sample</span>(<span class="fu">length</span>(S8))])]</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># log2 RPM normalize counts</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    sim.cnts.norm <span class="ot">=</span> sim.cnts[, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )), </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>        .SDcols <span class="ot">=</span> selcols]</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    sim.dtS86 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> sim.cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)],</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> sim.cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S8)])</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    sim.rraS86 <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">RRA</span>(sim.dtS86, <span class="at">enrichment =</span> T))</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    sim.rraS86.l <span class="ot">=</span> <span class="fu">c</span>(sim.rraS86.l, sim.rraS86)</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co"># get ecdf</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>ecdf.S86 <span class="ot">=</span> <span class="fu">ecdf</span>(<span class="fu">map</span>(sim.rraS86.l, <span class="sc">~</span>.x<span class="sc">$</span>rankscore) <span class="sc">%&gt;%</span> flatten <span class="sc">%&gt;%</span> unlist)</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co"># use ecdf to compute FDR value</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>rraS86[, FDR <span class="sc">:</span><span class="er">=</span> <span class="fu">ecdf.S86</span>(rankscore)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>N_sim <span class="ot">=</span> <span class="dv">20</span> <span class="co"># number of simulations</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>sim.rraS87.l <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N_sim) {</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    sim.cnts <span class="ot">=</span> cnts[</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>        , .(Gene, ID,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">S6 =</span> S6[<span class="fu">sample</span>(<span class="fu">length</span>(S6))],</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">S7 =</span> S7[<span class="fu">sample</span>(<span class="fu">length</span>(S7))],</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">S8 =</span> S8[<span class="fu">sample</span>(<span class="fu">length</span>(S8))])]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># log2 RPM normalize counts</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    sim.cnts.norm <span class="ot">=</span> sim.cnts[, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )), </span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>        .SDcols <span class="ot">=</span> selcols]</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    sim.dtS87 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> sim.cnts[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)],</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> sim.cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S7, <span class="at">num =</span> S8)])</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    sim.rraS87 <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">RRA</span>(sim.dtS87, <span class="at">enrichment =</span> T))</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    sim.rraS87.l <span class="ot">=</span> <span class="fu">c</span>(sim.rraS87.l, sim.rraS87)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co"># get ecdf</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>ecdf.S87 <span class="ot">=</span> <span class="fu">ecdf</span>(<span class="fu">map</span>(sim.rraS87.l, <span class="sc">~</span>.x<span class="sc">$</span>rankscore) <span class="sc">%&gt;%</span> flatten <span class="sc">%&gt;%</span> unlist)</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co"># use ecdf to compute FDR value</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>rraS87[, FDR <span class="sc">:</span><span class="er">=</span> <span class="fu">ecdf.S87</span>(rankscore)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>N_sim <span class="ot">=</span> <span class="dv">20</span> <span class="co"># number of simulations</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>sim.rraS76.l <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N_sim) {</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    sim.cnts <span class="ot">=</span> cnts[</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>        , .(Gene, ID,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">S6 =</span> S6[<span class="fu">sample</span>(<span class="fu">length</span>(S6))],</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">S7 =</span> S7[<span class="fu">sample</span>(<span class="fu">length</span>(S7))],</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">S8 =</span> S8[<span class="fu">sample</span>(<span class="fu">length</span>(S8))])]</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    selcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"S6"</span>, <span class="st">"S7"</span>, <span class="st">"S8"</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># log2 RPM normalize counts</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    sim.cnts.norm <span class="ot">=</span> sim.cnts[, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ID =</span> ID, <span class="at">Gene =</span> Gene),</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(.SD, <span class="cf">function</span>(x) <span class="fu">log2</span>(<span class="fu">if_else</span>(x<span class="sc">==</span><span class="dv">0</span>, .<span class="dv">1</span>, .<span class="dv">1</span><span class="sc">*</span>x)<span class="sc">*</span><span class="fl">1e6</span><span class="sc">/</span><span class="fu">sum</span>(x)) )), </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>        .SDcols <span class="ot">=</span> selcols]</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    sim.dtS76 <span class="ot">=</span> <span class="fu">makeDT</span>(<span class="at">dt =</span> sim.cnts[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)],</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">dt.norm =</span> sim.cnts.norm[, .(ID, Gene, <span class="at">denom =</span> S6, <span class="at">num =</span> S7)])</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    sim.rraS76 <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">RRA</span>(sim.dtS76, <span class="at">enrichment =</span> F))</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    sim.rraS76.l <span class="ot">=</span> <span class="fu">c</span>(sim.rraS76.l, sim.rraS76)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co"># get ecdf</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>ecdf.S76 <span class="ot">=</span> <span class="fu">ecdf</span>(<span class="fu">map</span>(sim.rraS76.l, <span class="sc">~</span>.x<span class="sc">$</span>rankscore) <span class="sc">%&gt;%</span> flatten <span class="sc">%&gt;%</span> unlist)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="co"># use ecdf to compute FDR value</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>rraS76[, FDR <span class="sc">:</span><span class="er">=</span> <span class="fu">ecdf.S76</span>(rankscore)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown" class="level3">
<h3 class="anchored" data-anchor-id="enrichment-of-td--vs.-in-at-day14-with-mc-computed-fdr-genes-with-fdr-0.15-are-shown">Enrichment of Td- vs.&nbsp;iN at Day14, with MC computed FDR, genes with FDR &lt; 0.15 are shown</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>rraS86[<span class="sc">!</span><span class="fu">is.na</span>(rankscore) <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> .<span class="dv">15</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                      Gene rankscore          FDR
  1:                ZNF441     28.75 0.0004593126
  2:                ZNF440     49.25 0.0021434586
  3:                ZBTB18     83.75 0.0067365842
  4:                  CRY1     88.00 0.0074255531
  5:                 BRWD1     88.25 0.0074255531
 ---                                             
100:                 RCOR2    411.25 0.1387889459
101: NonTargeting_Human_57    412.50 0.1399372273
102: NonTargeting_Human_73    417.25 0.1428462068
103:              MPHOSPH8    423.75 0.1472096762
104:                NBPF24    424.75 0.1479751971</code></pre>
</div>
</div>
</section>
<section id="enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr" class="level3">
<h3 class="anchored" data-anchor-id="enrichment-of-td--vs.-td-at-day14-with-mc-computed-fdr">Enrichment of Td- vs.&nbsp;Td+ at Day14, with MC computed FDR</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>rraS87[<span class="sc">!</span><span class="fu">is.na</span>(rankscore) <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> .<span class="dv">15</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                     Gene rankscore          FDR
 1:                ZNF441     10.75 0.0002468526
 2:                ZNF440     53.25 0.0032090842
 3:                  CRY1     54.25 0.0033736526
 4:                  E2F1     77.75 0.0072410105
 5:                  CDX4     81.25 0.0075701473
 6:               BHLHA15     84.50 0.0080638525
 7:                ZNF311     95.50 0.0099563894
 8:                  EGR3    105.50 0.0124249157
 9:                  ZIC1    107.75 0.0128363367
10:                 NR1I2    112.75 0.0143174525
11:                ZNF250    133.25 0.0189253682
12:                ZNF639    133.25 0.0189253682
13:                 PRDM6    135.00 0.0190899366
14:                  DLX5    136.75 0.0195836419
15:                  TOX4    139.75 0.0203241998
16:                  THRA    140.00 0.0203241998
17:                 BRWD1    143.50 0.0212293261
18:                  CHD5    145.00 0.0214761787
19:                  CBX4    162.50 0.0272360734
20:                  VAX2    162.50 0.0272360734
21:                ZBTB18    164.75 0.0277297787
22:                 RAI14    165.00 0.0278120629
23:                  RBPJ    166.25 0.0279766313
24:                 TSHZ3    171.00 0.0293754628
25:                NKX3-2    173.75 0.0301160207
26:                 TAF1L    176.75 0.0311857155
27:                 APBB2    178.50 0.0320908418
28:                 SCRT2    180.00 0.0327491154
29:                 HSFY1    183.25 0.0336542418
30:                ZNF136    190.00 0.0365341891
31:                  EMX1    197.75 0.0395787048
32:                 UBE2K    197.75 0.0395787048
33:                 CTDP1    204.25 0.0426232206
34:                  JUNB    206.25 0.0436106311
35:                 MECP2    209.25 0.0445980416
36:                POLR2B    210.00 0.0450917469
37:                  ZZZ3    210.00 0.0450917469
38: NonTargeting_Human_22    212.50 0.0459968732
39:                ZNF554    214.00 0.0469842837
40:                 KLF10    220.50 0.0498642311
41:                 INSM1    229.50 0.0541430100
42:                 MEOX2    232.00 0.0552127047
43:                ZNF671    232.00 0.0552127047
44: NonTargeting_Human_90    236.50 0.0571875257
45:                 SMAD3    239.00 0.0586686415
46:                  EBF1    240.00 0.0594091994
47:                ZNF563    250.25 0.0644285362
48:                  PAX7    252.00 0.0654159467
49:                ZNF141    253.00 0.0656627993
50:                 ZFPM1    258.75 0.0677199046
51:             HIST1H2BC    264.00 0.0701061466
52:                  JUND    267.75 0.0717518308
53:               HTATIP2    268.75 0.0719986835
54:                 PRDM5    280.25 0.0781699992
55:                 GATA1    295.00 0.0852464412
56:                  TFEB    296.50 0.0863161359
57:                CREBZF    297.50 0.0871389780
58:                 VGLL2    297.75 0.0871389780
59:                   SKI    298.00 0.0873858307
60:                   ZFY    298.00 0.0873858307
61:                 HIF3A    301.75 0.0893606517
62: NonTargeting_Human_13    306.00 0.0919937464
63:                  ELK3    318.00 0.0982473463
64: NonTargeting_Human_31    319.00 0.0990701884
65: NonTargeting_Human_57    319.25 0.0991524726
66:                  ZIM3    320.25 0.0995638937
67:                ARID1B    323.50 0.1013741463
68:                   AHR    324.75 0.1017032831
69: NonTargeting_Human_88    325.25 0.1019501358
70:                 NFAT5    328.50 0.1042540936
71:               NEUROG3    333.25 0.1077923147
72:               TSC22D3    335.50 0.1085328725
73:                  LHX4    341.00 0.1114951041
74: NonTargeting_Human_69    341.75 0.1119888093
75:                ZNF239    343.25 0.1130585041
76: NonTargeting_Human_39    346.75 0.1145396198
77:                  SIX5    347.25 0.1147041883
78:                ZNF510    355.25 0.1193943882
79:                  PAX2    357.00 0.1199703777
80:                 ZNF92    362.50 0.1232617461
81:                CTNNB1    363.75 0.1239200197
82:                  TAF5    365.25 0.1247428618
83:                NFKBIB    368.00 0.1267999671
84: NonTargeting_Human_99    369.75 0.1271291039
85:               ZKSCAN3    380.00 0.1333827039
86:                  AFF4    380.75 0.1345346828
87:                 ATF6B    381.00 0.1345346828
88:                 ARNTL    381.25 0.1346992512
89:                 TCEA2    382.25 0.1354398091
90:                 PHF12    384.50 0.1370854933
91:              MPHOSPH8    385.25 0.1374969143
92:                 ZNF34    390.75 0.1407882827
93:                  RORB    397.25 0.1446556406
94:                 HOXB3    399.25 0.1468773142
                     Gene rankscore          FDR</code></pre>
</div>
</div>
</section>
<section id="depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr" class="level3">
<h3 class="anchored" data-anchor-id="depletion-of-td-vs.-in-at-day14-with-mc-computed-fdr">Depletion of Td+ vs.&nbsp;iN at Day14, with MC computed FDR</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>rraS76[FDR <span class="sc">&lt;</span> .<span class="dv">15</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>       Gene rankscore          FDR
  1:   MXI1     62.50 0.0005569620
  2:  NCOA3     65.00 0.0006329114
  3:  NR5A2     70.25 0.0008101266
  4:   TFEC     84.00 0.0011392405
  5: HIF1AN     87.50 0.0011645570
 ---                              
306:  NFKB1   1176.50 0.1460253165
307:   TBX5   1176.50 0.1460253165
308: ANKRD1   1177.75 0.1462278481
309: ZNF229   1178.25 0.1463291139
310:    SP4   1193.25 0.1493417722</code></pre>
</div>
</div>
<hr>
</section>
</section>
<section id="compare-with-hits-found-from-existing-method-using-fdr-.15" class="level2">
<h2 class="anchored" data-anchor-id="compare-with-hits-found-from-existing-method-using-fdr-.15">Compare with hits found from existing method, using FDR&lt; .15</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>hitlist <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"../code/resources/crispr/Hitlist_20191230.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Previous method found 122 genes, shown below:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>  [1] "PHOX2A"    "TAF1L"     "TFEB"      "VAX2"      "SNAI2"     "ELK3"     
  [7] "ISL2"      "ZBTB43"    "ZNF200"    "MLLT6"     "NFAT5"     "ZNF563"   
 [13] "ZNF3"      "ZFY"       "ZNF502"    "RCOR2"     "TSC22D3"   "NR1I2"    
 [19] "RBPJ"      "HSFY1"     "TAF5"      "RAI14"     "PHF12"     "BRWD1"    
 [25] "FOXQ1"     "CREBZF"    "EMX1"      "ZNF324"    "ZIM3"      "ZNF311"   
 [31] "ZNF136"    "EGR3"      "ASB1"      "CBX4"      "GTF2A2"    "ZKSCAN3"  
 [37] "HOXC13"    "ZNF671"    "ZNF639"    "KLF10"     "JADE3"     "ARID1B"   
 [43] "ZNF547"    "CNOT6"     "EBF1"      "AHR"       "PAX7"      "POLR2B"   
 [49] "SIX5"      "CDX4"      "GATA1"     "ZNF573"    "NKX3-2"    "ZNF250"   
 [55] "POU2F3"    "PRDM5"     "NBPF24"    "ZNF34"     "SKI"       "E2F1"     
 [61] "ARNTL"     "MEOX2"     "AIP"       "ZNF510"    "ZNF440"    "DLX5"     
 [67] "HIF3A"     "SOX15"     "APBB2"     "TOX4"      "ZNF239"    "FOXN2"    
 [73] "SCRT2"     "RORB"      "CRY1"      "ZNF92"     "JUND"      "CEBPA"    
 [79] "INSM1"     "ZNF441"    "THRA"      "NKRF"      "MPHOSPH8"  "NR1D2"    
 [85] "ZFPM1"     "JUNB"      "TBPL1"     "ZNF141"    "BHLHA15"   "EHF"      
 [91] "ZZZ3"      "ZIC1"      "POU3F2"    "HOXB3"     "KLF14"     "CTDP1"    
 [97] "VGLL2"     "AFF4"      "ZNF282"    "MECP2"     "TCEA2"     "SLA2"     
[103] "TSHZ3"     "NFKBIB"    "PAX2"      "POLR2J"    "ZFP2"      "ZNF107"   
[109] "ZBTB18"    "CHD5"      "CTNNB1"    "ZNF554"    "UBE2K"     "HTATIP2"  
[115] "NEUROG3"   "SMAD3"     "LHX4"      "PRDM6"     "NEUROG1"   "NEUROG2"  
[121] "HIST1H2BC" "HIST1H1E" </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"venn.svg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="venn.svg" class="img-fluid"></p>
</div>
</div>
<p>Notice a large number of overlaps. When we combine hits found from Td- vs.&nbsp;iN and Td- vs.&nbsp;Td+ with FDR &lt; 0.15 (which is the same way we selected our 120 hits. There are 23 (NEUROG1 and NEUROG2 are not considered) hits found in existing method, but not found in the RRA method. Nearly all the hits that we initially removed in our existing method, which are labeled as hit with the RRA method are non-targeting controls. This highlights the advantage of our existing manual method of more precise control of false positives and potential false negatives. Of course, the disadvantage is that the existing manual method does not produce a p-value thus is less informative in statistical confidence.</p>
<p>Below shows hits that are found in both this method and our existing method. Notice many of these TFs are repeatedly identified as differentially expressed as well as later identified as nodes in the putative neuronal development gene regulatory network.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">intersect</span>(<span class="fu">union</span>(set87, set86), hitlist<span class="sc">$</span>gene_name) <span class="sc">%&gt;%</span> length</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 97</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">intersect</span>(<span class="fu">union</span>(set87, set86), hitlist<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ZNF441"    "ZNF440"    "CRY1"      "E2F1"      "CDX4"      "BHLHA15"  
 [7] "ZNF311"    "EGR3"      "ZIC1"      "NR1I2"     "ZNF250"    "ZNF639"   
[13] "PRDM6"     "DLX5"      "TOX4"      "THRA"      "BRWD1"     "CHD5"     
[19] "CBX4"      "VAX2"      "ZBTB18"    "RAI14"     "RBPJ"      "TSHZ3"    
[25] "NKX3-2"    "TAF1L"     "APBB2"     "SCRT2"     "HSFY1"     "ZNF136"   
[31] "EMX1"      "UBE2K"     "CTDP1"     "JUNB"      "MECP2"     "POLR2B"   
[37] "ZZZ3"      "ZNF554"    "KLF10"     "INSM1"     "MEOX2"     "ZNF671"   
[43] "SMAD3"     "EBF1"      "ZNF563"    "PAX7"      "ZNF141"    "ZFPM1"    
[49] "HIST1H2BC" "JUND"      "HTATIP2"   "PRDM5"     "GATA1"     "TFEB"     
[55] "CREBZF"    "VGLL2"     "SKI"       "ZFY"       "HIF3A"     "ELK3"     
[61] "ZIM3"      "ARID1B"    "AHR"       "NFAT5"     "NEUROG3"   "TSC22D3"  
[67] "LHX4"      "ZNF239"    "SIX5"      "ZNF510"    "PAX2"      "ZNF92"    
[73] "CTNNB1"    "TAF5"      "NFKBIB"    "ZKSCAN3"   "AFF4"      "ARNTL"    
[79] "TCEA2"     "PHF12"     "MPHOSPH8"  "ZNF34"     "RORB"      "HOXB3"    
[85] "ISL2"      "ZNF502"    "FOXQ1"     "HOXC13"    "POU2F3"    "POLR2J"   
[91] "TBPL1"     "AIP"       "ZNF282"    "CNOT6"     "ZNF547"    "RCOR2"    
[97] "NBPF24"   </code></pre>
</div>
</div>
<p>Below are hits that are found in our existing method but missed in this method.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(hitlist<span class="sc">$</span>gene_name, <span class="fu">union</span>(set87, set86))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "PHOX2A"   "SNAI2"    "ZBTB43"   "ZNF200"   "MLLT6"    "ZNF3"    
 [7] "ZNF324"   "ASB1"     "GTF2A2"   "JADE3"    "ZNF573"   "SOX15"   
[13] "FOXN2"    "CEBPA"    "NKRF"     "NR1D2"    "EHF"      "POU3F2"  
[19] "KLF14"    "SLA2"     "ZFP2"     "ZNF107"   "NEUROG1"  "NEUROG2" 
[25] "HIST1H1E"</code></pre>
</div>
</div>
<p>Below are labeled hits in this method but rejected in our existing method. Notice many are in fact non-targeting controls, suggesting our existing method may be better suited in removing false positives.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">union</span>(set87, set86), hitlist<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "NonTargeting_Human_22" "NonTargeting_Human_90" "NonTargeting_Human_13"
 [4] "NonTargeting_Human_31" "NonTargeting_Human_57" "NonTargeting_Human_88"
 [7] "NonTargeting_Human_69" "NonTargeting_Human_39" "NonTargeting_Human_99"
[10] "ATF6B"                 "NonTargeting_Human_67" "ZNF766"               
[13] "NonTargeting_Human_64" "NonTargeting_Human_73"</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>