<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.280">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Chao Dai">
<meta name="dcterms.date" content="2022-07-28">

<title>NeuronReporters - QC</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">NeuronReporters</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#library-description" id="toc-library-description" class="nav-link active" data-scroll-target="#library-description">Library Description</a></li>
  <li><a href="#crispr" id="toc-crispr" class="nav-link" data-scroll-target="#crispr">CRISPR</a>
  <ul class="collapse">
  <li><a href="#numer-of-guides" id="toc-numer-of-guides" class="nav-link" data-scroll-target="#numer-of-guides">numer of guides</a></li>
  </ul></li>
  <li><a href="#hits" id="toc-hits" class="nav-link" data-scroll-target="#hits">Hits</a></li>
  <li><a href="#number-of-atac-seq-reads-in-promoters" id="toc-number-of-atac-seq-reads-in-promoters" class="nav-link" data-scroll-target="#number-of-atac-seq-reads-in-promoters">Number of ATAC-seq reads in promoters</a></li>
  <li><a href="#number-of-atac-seeq-raw-sequenced-reads" id="toc-number-of-atac-seeq-raw-sequenced-reads" class="nav-link" data-scroll-target="#number-of-atac-seeq-raw-sequenced-reads">Number of ATAC-seeq raw sequenced reads</a></li>
  <li><a href="#number-of-rna-seq-raw-sequenced-reads" id="toc-number-of-rna-seq-raw-sequenced-reads" class="nav-link" data-scroll-target="#number-of-rna-seq-raw-sequenced-reads">Number of RNA-seq raw sequenced reads</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">QC</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Chao Dai </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 28, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="library-description" class="level2">
<h2 class="anchored" data-anchor-id="library-description">Library Description</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># config yaml</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>config <span class="ot">=</span> <span class="fu">yaml.load_file</span>(<span class="st">"../code/config/config.yaml"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sample sheets</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>ss <span class="ot">=</span> <span class="fu">list</span>(<span class="at">rna =</span> <span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">"../code/"</span>, config<span class="sc">$</span>RNA_SAMPLES)),</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">atac =</span> <span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">"../code/"</span>, config<span class="sc">$</span>ATAC_SAMPLES))</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>          )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ss<span class="sc">$</span>rna <span class="ot">=</span> ss<span class="sc">$</span>rna[, .(</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">library =</span> <span class="st">"rna-seq"</span>,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="fu">if_else</span>(batch <span class="sc">==</span> <span class="st">"A"</span>, <span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">sampleid =</span> sample,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">paste</span>(treatment, time, <span class="at">sep=</span><span class="st">"-"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>)] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>ss<span class="sc">$</span>atac <span class="ot">=</span> ss<span class="sc">$</span>atac[, .(</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">library =</span> <span class="st">"atac-seq"</span>, </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="dv">1</span>,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">sampleid =</span> <span class="fu">paste</span>(<span class="st">"S"</span>, <span class="fu">str_match</span>(fastq, <span class="st">"ATAC([0-9]{1,2})"</span>)[, <span class="dv">2</span>], <span class="at">sep=</span><span class="st">""</span>),</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">paste</span>(<span class="st">"wt"</span>, time, <span class="at">sep=</span><span class="st">"-"</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>)] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>ss <span class="ot">=</span> <span class="fu">rbindlist</span>(ss)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co"># bam flagstats</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>flagstat <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"../code/results/QC/flagstat"</span>, <span class="st">'*txt'</span>, <span class="at">recursive =</span> T, <span class="at">full.names =</span> T)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>flagstat_meta <span class="ot">=</span> <span class="fu">str_match</span>(flagstat, <span class="st">"([rnatc]{3,4}-seq).+(batch[1/2]).+(S[0-9]{2})"</span>)  <span class="sc">%&gt;%</span> as.data.table</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>flagstat_meta <span class="ot">=</span> flagstat_meta[, .(</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">library =</span> V2,</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="fu">str_extract</span>(V3, <span class="st">"[0-9]{1}"</span>) <span class="sc">%&gt;%</span> as.numeric,</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">sampleid =</span> V4,</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> flagstat</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>)] <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="co"># join</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>flagstat_meta <span class="ot">=</span> ss[flagstat_meta, on <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"library"</span>, <span class="st">"batch"</span>, <span class="st">"sampleid"</span>), nomatch <span class="ot">=</span> <span class="cn">NA</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library size = primary mapped reads</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>flagstat_meta[, data <span class="sc">:</span><span class="er">=</span> <span class="fu">map</span>(file, fread)]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>lib_size1 <span class="ot">=</span> flagstat_meta[, .(</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">description =</span> <span class="fu">paste</span>(library, condition, <span class="at">sep =</span> <span class="st">"::"</span>),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  library,</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  batch,</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  sampleid,</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  condition,</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">primary_mapped =</span> <span class="fu">map_dbl</span>(data, <span class="sc">~</span> .x<span class="sc">$</span>V1[[<span class="dv">9</span>]] <span class="sc">%&gt;%</span> as.numeric)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>)][<span class="fu">order</span>(description), .(<span class="at">size =</span> <span class="fu">sum</span>(primary_mapped)), by <span class="ot">=</span> <span class="st">"description"</span>] </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>lib_size2 <span class="ot">=</span> flagstat_meta[, .(</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">description =</span> <span class="fu">paste</span>(library, condition, <span class="at">sep =</span> <span class="st">"::"</span>),</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  library,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  batch,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  sampleid,</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  condition,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">primary_mapped =</span> <span class="fu">map_dbl</span>(data, <span class="sc">~</span> .x<span class="sc">$</span>V1[[<span class="dv">9</span>]] <span class="sc">%&gt;%</span> as.numeric)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>)][, .(<span class="at">size =</span> <span class="fu">sum</span>(primary_mapped)), by <span class="ot">=</span> <span class="st">"library"</span>]</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>lib_size3 <span class="ot">=</span> flagstat_meta[, .(</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  library,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  batch,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  sampleid,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  condition,</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">primary_mapped =</span> <span class="fu">map_dbl</span>(data, <span class="sc">~</span> .x<span class="sc">$</span>V1[[<span class="dv">9</span>]] <span class="sc">%&gt;%</span> as.numeric)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>)][<span class="fu">order</span>(library, condition, sampleid), .(<span class="at">size =</span> <span class="fu">sum</span>(primary_mapped)), by <span class="ot">=</span> .(library, condition, sampleid)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Overall we generated 602,272,627 primarily mapped reads for the ATAC-seq library and 273,173,427 primarily mapped reads for the RNA-seq library ( table @ref(tab:table1).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># total library size</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>lib_size2[, .(library, <span class="at">size =</span> scales<span class="sc">::</span><span class="fu">comma</span>(size))] <span class="co">#%&gt;% knitr::kable(., caption = "Total library size")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>    library        size
1: atac-seq 602,272,627
2:  rna-seq 273,173,427</code></pre>
</div>
</div>
<p>Detailed library size in table @ref(tab:table2)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library size by condition</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>lib_size1[, .(description,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">size =</span> scales<span class="sc">::</span><span class="fu">comma</span>(size)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>                )] </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         description        size
 1:  atac-seq::wt-d1 145,203,102
 2:  atac-seq::wt-d5  86,612,423
 3:  atac-seq::wt-es  89,820,545
 4:  atac-seq::wt-h1  93,772,813
 5: atac-seq::wt-h16  82,710,864
 6:  atac-seq::wt-h4 104,152,880
 7: rna-seq::k108-d4  21,564,934
 8: rna-seq::k108-es  22,591,656
 9: rna-seq::k112-d4  24,190,413
10: rna-seq::k112-es  33,760,481
11:   rna-seq::wt-d1  25,727,105
12:   rna-seq::wt-d2  25,784,430
13:   rna-seq::wt-d4  21,655,783
14:   rna-seq::wt-d7  19,337,421
15:   rna-seq::wt-es  52,242,551
16:  rna-seq::wt-h15  26,318,653</code></pre>
</div>
</div>
<p><strong>sample</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>lib_size3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     library condition sampleid     size
 1: atac-seq     wt-d1      S09 86433463
 2: atac-seq     wt-d1      S10 58769639
 3: atac-seq     wt-d5      S11 45593714
 4: atac-seq     wt-d5      S12 41018709
 5: atac-seq     wt-es      S01 47160567
 6: atac-seq     wt-es      S02 42659978
 7: atac-seq     wt-h1      S03 47790998
 8: atac-seq     wt-h1      S04 45981815
 9: atac-seq    wt-h16      S07 39407650
10: atac-seq    wt-h16      S08 43303214
11: atac-seq     wt-h4      S05 42795268
12: atac-seq     wt-h4      S06 61357612
13:  rna-seq   k108-d4      S09 13009210
14:  rna-seq   k108-d4      S10  8555724
15:  rna-seq   k108-es      S03  9575107
16:  rna-seq   k108-es      S04 13016549
17:  rna-seq   k112-d4      S11  9805283
18:  rna-seq   k112-d4      S12 14385130
19:  rna-seq   k112-es      S05 14726059
20:  rna-seq   k112-es      S06 19034422
21:  rna-seq     wt-d1      S17 12845991
22:  rna-seq     wt-d1      S18 12881114
23:  rna-seq     wt-d2      S19 12064070
24:  rna-seq     wt-d2      S20 13720360
25:  rna-seq     wt-d4      S07  7307671
26:  rna-seq     wt-d4      S08 14348112
27:  rna-seq     wt-d7      S21  8518963
28:  rna-seq     wt-d7      S22 10818458
29:  rna-seq     wt-es      S01 11492703
30:  rna-seq     wt-es      S02 12942704
31:  rna-seq     wt-es      S13 14631896
32:  rna-seq     wt-es      S14 13175248
33:  rna-seq    wt-h15      S15 13690829
34:  rna-seq    wt-h15      S16 12627824
     library condition sampleid     size</code></pre>
</div>
</div>
<p><strong>by sample and batch</strong></p>
<p>ATAC</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>flagstat_meta[, .(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  library,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  batch,</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  sampleid,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  condition,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">primary_mapped =</span> <span class="fu">map_dbl</span>(data, <span class="sc">~</span> .x<span class="sc">$</span>V1[[<span class="dv">9</span>]] <span class="sc">%&gt;%</span> <span class="fu">as.integer</span>(.))</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>)][<span class="fu">order</span>(library, condition, sampleid, batch), .(<span class="at">size =</span> <span class="fu">sum</span>(primary_mapped)), by <span class="ot">=</span> .(library, condition, sampleid, batch)] <span class="sc">%&gt;%</span> </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  .[library <span class="sc">==</span> <span class="st">"atac-seq"</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">library</th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">sampleid</th>
<th style="text-align: right;">batch</th>
<th style="text-align: right;">size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-d1</td>
<td style="text-align: left;">S09</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">86433463</td>
</tr>
<tr class="even">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-d1</td>
<td style="text-align: left;">S10</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">58769639</td>
</tr>
<tr class="odd">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-d5</td>
<td style="text-align: left;">S11</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">45593714</td>
</tr>
<tr class="even">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-d5</td>
<td style="text-align: left;">S12</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">41018709</td>
</tr>
<tr class="odd">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S01</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">47160567</td>
</tr>
<tr class="even">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S02</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">42659978</td>
</tr>
<tr class="odd">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-h1</td>
<td style="text-align: left;">S03</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">47790998</td>
</tr>
<tr class="even">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-h1</td>
<td style="text-align: left;">S04</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">45981815</td>
</tr>
<tr class="odd">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-h16</td>
<td style="text-align: left;">S07</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">39407650</td>
</tr>
<tr class="even">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-h16</td>
<td style="text-align: left;">S08</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">43303214</td>
</tr>
<tr class="odd">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-h4</td>
<td style="text-align: left;">S05</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">42795268</td>
</tr>
<tr class="even">
<td style="text-align: left;">atac-seq</td>
<td style="text-align: left;">wt-h4</td>
<td style="text-align: left;">S06</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">61357612</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>RNA</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>flagstat_meta[, .(</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  library,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  batch,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  sampleid,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  condition,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">primary_mapped =</span> <span class="fu">map_dbl</span>(data, <span class="sc">~</span> .x<span class="sc">$</span>V1[[<span class="dv">9</span>]] <span class="sc">%&gt;%</span> <span class="fu">as.integer</span>(.))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>)][<span class="fu">order</span>(library, condition, sampleid, batch), .(<span class="at">size =</span> <span class="fu">sum</span>(primary_mapped)), by <span class="ot">=</span> .(library, condition, sampleid, batch)] <span class="sc">%&gt;%</span> </span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  .[library <span class="sc">==</span> <span class="st">"rna-seq"</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">library</th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">sampleid</th>
<th style="text-align: right;">batch</th>
<th style="text-align: right;">size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-d4</td>
<td style="text-align: left;">S09</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6055433</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-d4</td>
<td style="text-align: left;">S09</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">6953777</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-d4</td>
<td style="text-align: left;">S10</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2826607</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-d4</td>
<td style="text-align: left;">S10</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5729117</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-es</td>
<td style="text-align: left;">S03</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5163810</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-es</td>
<td style="text-align: left;">S03</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">4411297</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-es</td>
<td style="text-align: left;">S04</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5059043</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k108-es</td>
<td style="text-align: left;">S04</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7957506</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-d4</td>
<td style="text-align: left;">S11</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4475065</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-d4</td>
<td style="text-align: left;">S11</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5330218</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-d4</td>
<td style="text-align: left;">S12</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5894307</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-d4</td>
<td style="text-align: left;">S12</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">8490823</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-es</td>
<td style="text-align: left;">S05</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6624411</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-es</td>
<td style="text-align: left;">S05</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">8101648</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-es</td>
<td style="text-align: left;">S06</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6243150</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">k112-es</td>
<td style="text-align: left;">S06</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">12791272</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d1</td>
<td style="text-align: left;">S17</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5066315</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d1</td>
<td style="text-align: left;">S17</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7779676</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d1</td>
<td style="text-align: left;">S18</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5014762</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d1</td>
<td style="text-align: left;">S18</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7866352</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d2</td>
<td style="text-align: left;">S19</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5723479</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d2</td>
<td style="text-align: left;">S19</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">6340591</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d2</td>
<td style="text-align: left;">S20</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5691175</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d2</td>
<td style="text-align: left;">S20</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">8029185</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d4</td>
<td style="text-align: left;">S07</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5327393</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d4</td>
<td style="text-align: left;">S07</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1980278</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d4</td>
<td style="text-align: left;">S08</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">7052039</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d4</td>
<td style="text-align: left;">S08</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7296073</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d7</td>
<td style="text-align: left;">S21</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3702690</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d7</td>
<td style="text-align: left;">S21</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">4816273</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d7</td>
<td style="text-align: left;">S22</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4173331</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-d7</td>
<td style="text-align: left;">S22</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">6645127</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S01</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5625097</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S01</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5867606</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S02</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5704710</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S02</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7237994</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S13</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6621604</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S13</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">8010292</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S14</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5398943</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-es</td>
<td style="text-align: left;">S14</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7776305</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-h15</td>
<td style="text-align: left;">S15</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6157027</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-h15</td>
<td style="text-align: left;">S15</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">7533802</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-h15</td>
<td style="text-align: left;">S16</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5709784</td>
</tr>
<tr class="even">
<td style="text-align: left;">rna-seq</td>
<td style="text-align: left;">wt-h15</td>
<td style="text-align: left;">S16</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">6918040</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="crispr" class="level2">
<h2 class="anchored" data-anchor-id="crispr">CRISPR</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>cnts.file <span class="ot">=</span> <span class="st">"../code/resources/crispr/counts/wrangled_raw_counts.tsv"</span> <span class="co"># raw file</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>cnts <span class="ot">=</span> <span class="fu">fread</span>(cnts.file) <span class="sc">%&gt;%</span> .[, .(<span class="at">ID=</span>sgRNA, Gene, <span class="at">S6 =</span> Day14_iN, <span class="at">S7=</span>Day14_Tdpos, <span class="at">S8=</span>Day14_Tdneg)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="numer-of-guides" class="level3">
<h3 class="anchored" data-anchor-id="numer-of-guides">numer of guides</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>cnts[, type <span class="sc">:</span><span class="er">=</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(Gene, <span class="st">"NonTarget"</span>), <span class="st">"NTC"</span>, <span class="st">"Targeting"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>cnts[, .(<span class="at">N_total_gRNAs =</span> <span class="fu">length</span>(<span class="fu">unique</span>(ID))), by <span class="ot">=</span> type]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>        type N_total_gRNAs
1: Targeting         18838
2:       NTC          1000</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>cnts[ type <span class="sc">==</span> <span class="st">"Targeting"</span>, Gene] <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> length</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1891</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># group 10 NTC guides into 1 NTC group</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>NTC <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"NonTargeting_Human_"</span>, <span class="fu">sort</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">10</span>)), <span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>cnts[<span class="fu">str_detect</span>(<span class="fu">toupper</span>(Gene), <span class="st">"NONTARGET"</span>), Gene <span class="sc">:</span><span class="er">=</span> NTC]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>total library size</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>cnts[, .SD, .SDcols <span class="ot">=</span> is.numeric] <span class="sc">%&gt;%</span> sum</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4769269</code></pre>
</div>
</div>
<p>size by library</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>cnts[, .SD, .SDcols <span class="ot">=</span> is.numeric] <span class="sc">%&gt;%</span> colSums</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     S6      S7      S8 
1717607 1290124 1761538 </code></pre>
</div>
</div>
</section>
</section>
<section id="hits" class="level2">
<h2 class="anchored" data-anchor-id="hits">Hits</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>hitlist <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"../code/resources/crispr/Hitlist_20191230.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> cnts[, .(Gene, ID, S6, S7, S8, <span class="at">rs =</span> <span class="fu">rowSums</span>(cnts[, .(S6, S7, S8)]))][</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  Gene <span class="sc">%in%</span> hitlist<span class="sc">$</span>gene_name <span class="sc">&amp;</span> rs<span class="sc">&gt;</span><span class="dv">0</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>][<span class="fu">order</span>(Gene, rs), <span class="sc">-</span><span class="fu">c</span>(<span class="st">"rs"</span>)]</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> dt[, .(Gene, ID, <span class="at">iN =</span> S6, <span class="at">td_pos =</span> S7, <span class="at">td_neg =</span> S8)]</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(dt, <span class="st">"../data/hitTFs-reads-by-gRNA.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>dt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>      Gene             ID iN td_pos td_neg
   1: AFF4 GUIDES_sg00114  3      0      0
   2: AFF4 GUIDES_sg00111  6      6      0
   3: AFF4 GUIDES_sg00113 26      7      0
   4: AFF4 GUIDES_sg00115 43     14      0
   5: AFF4 GUIDES_sg00112 50     40      0
  ---                                     
1150: ZZZ3 GUIDES_sg18836 20     17      0
1151: ZZZ3 GUIDES_sg18832 34     12      0
1152: ZZZ3 GUIDES_sg18835 59     49      0
1153: ZZZ3 GUIDES_sg18833 38     36   1557
1154: ZZZ3 GUIDES_sg18829 80    118   2257</code></pre>
</div>
</div>
</section>
<section id="number-of-atac-seq-reads-in-promoters" class="level2">
<h2 class="anchored" data-anchor-id="number-of-atac-seq-reads-in-promoters">Number of ATAC-seq reads in promoters</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>preads.f <span class="ot">&lt;-</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"../code/results/reviews/atac/counts/promoter/ATAC{1:12}.txt.summary"</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(preads.f) <span class="ot">&lt;-</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"ATAC{1:12}"</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>preads <span class="ot">&lt;-</span> <span class="fu">map</span>(preads.f, \(x) <span class="fu">fread</span>(x) <span class="sc">%&gt;%</span> deframe <span class="sc">%&gt;%</span> .[[<span class="st">"Assigned"</span>]]) <span class="sc">%&gt;%</span> unlist</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>preads <span class="ot">&lt;-</span> <span class="fu">enframe</span>(preads)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>preads <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sampleid =</span> <span class="fu">str_replace</span>(name, <span class="st">"ATAC"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> <span class="fu">str_pad</span>(<span class="dv">2</span>, <span class="at">pad=</span><span class="st">"0"</span>) <span class="sc">%&gt;%</span> <span class="fu">str_pad</span>(<span class="dv">3</span>, <span class="at">pad=</span><span class="st">"S"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">filter</span>(ss, library <span class="sc">==</span> <span class="st">"atac-seq"</span>), <span class="at">by =</span> <span class="st">"sampleid"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library, condition, sampleid, value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 12 × 4
   library  condition sampleid   value
   &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;      &lt;int&gt;
 1 atac-seq wt-es     S01       776638
 2 atac-seq wt-es     S02       676624
 3 atac-seq wt-h1     S03       700464
 4 atac-seq wt-h1     S04       689459
 5 atac-seq wt-h4     S05       726751
 6 atac-seq wt-h4     S06       989551
 7 atac-seq wt-h16    S07       601360
 8 atac-seq wt-h16    S08       824131
 9 atac-seq wt-d1     S09      1521868
10 atac-seq wt-d1     S10      1014683
11 atac-seq wt-d5     S11      1209667
12 atac-seq wt-d5     S12      1080010</code></pre>
</div>
</div>
</section>
<section id="number-of-atac-seeq-raw-sequenced-reads" class="level2">
<h2 class="anchored" data-anchor-id="number-of-atac-seeq-raw-sequenced-reads">Number of ATAC-seeq raw sequenced reads</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC01_11-14-18_fastqc/fastqc_data.txt</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>fastqdata <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC{str_pad(1:12, 2, pad=</span><span class="sc">\'</span><span class="st">0</span><span class="sc">\'</span><span class="st">)}_11-14-18_fastqc/fastqc_data.txt"</span>),</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC{str_pad(1:12, 2, pad=</span><span class="sc">\'</span><span class="st">0</span><span class="sc">\'</span><span class="st">)}_11-14-18_fastqc/fastqc_data.txt"</span>)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>cmds <span class="ot">&lt;-</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"awk '$0 ~ /Total Sequences/ {{print $0}}' {fastqdata}"</span>)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cmds) <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(cmds, <span class="st">"(n</span><span class="sc">\\</span><span class="st">d{2}).+(ATAC</span><span class="sc">\\</span><span class="st">d{2})"</span>)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(cmds, \(x) <span class="fu">fread</span>(x) <span class="sc">%&gt;%</span> .[, V2]) <span class="sc">%&gt;%</span> unlist <span class="sc">%&gt;%</span> enframe <span class="sc">%&gt;%</span> </span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(name, <span class="fu">c</span>(<span class="st">"read"</span>, <span class="st">"sampleid"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sampleid =</span> <span class="fu">str_replace</span>(sampleid, <span class="st">"ATAC"</span>, <span class="st">"S"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sampleid) <span class="sc">%&gt;%</span> </span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">readsbysample =</span> <span class="fu">sum</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(ss[library <span class="sc">==</span> <span class="st">"atac-seq"</span>, .(library, sampleid, condition)] <span class="sc">%&gt;%</span> unique, <span class="at">by =</span> <span class="st">"sampleid"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library, condition, sampleid, readsbysample) <span class="sc">%&gt;%</span> </span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">readsbysample =</span> scales<span class="sc">::</span><span class="fu">number</span>(readsbysample)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sampleid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC01_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC02_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC03_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC04_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC05_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC06_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC07_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC08_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC09_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC10_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC11_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n01_ATAC12_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC01_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC02_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC03_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC04_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC05_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC06_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC07_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC08_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC09_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC10_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC11_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Taking input= as a system command ('awk '$0 ~ /Total Sequences/ {print $0}' ../code/results/ATACseq/fastqc/HG7FNBGX9_n02_ATAC12_11-14-18_fastqc/fastqc_data.txt') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 12 × 4
   library  condition sampleid readsbysample
   &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;    &lt;chr&gt;        
 1 atac-seq wt-es     S01      59 888 540   
 2 atac-seq wt-es     S02      53 791 264   
 3 atac-seq wt-h1     S03      61 414 542   
 4 atac-seq wt-h1     S04      58 651 844   
 5 atac-seq wt-h4     S05      54 478 534   
 6 atac-seq wt-h4     S06      79 064 076   
 7 atac-seq wt-h16    S07      50 592 262   
 8 atac-seq wt-h16    S08      54 346 242   
 9 atac-seq wt-d1     S09      110 710 086  
10 atac-seq wt-d1     S10      74 905 594   
11 atac-seq wt-d5     S11      57 281 292   
12 atac-seq wt-d5     S12      51 539 826   </code></pre>
</div>
</div>
</section>
<section id="number-of-rna-seq-raw-sequenced-reads" class="level2">
<h2 class="anchored" data-anchor-id="number-of-rna-seq-raw-sequenced-reads">Number of RNA-seq raw sequenced reads</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ../code/results/RNAseq/fastqc/HFJM7BGXC_n01_S17_fastqc/fastqc_data.txt</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="co">#HFJM7BGXC_n01_S19_fastqc.zip</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>fastqdata <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"../code/results/RNAseq/fastqc/H2HY7BGXC_n01_S{str_pad(1:22, 2, pad=</span><span class="sc">\'</span><span class="st">0</span><span class="sc">\'</span><span class="st">)}_fastqc/fastqc_data.txt"</span>),</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"../code/results/RNAseq/fastqc/HFJM7BGXC_n01_S{str_pad(1:22, 2, pad=</span><span class="sc">\'</span><span class="st">0</span><span class="sc">\'</span><span class="st">)}_fastqc/fastqc_data.txt"</span>)</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>batches <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"H2HY7BGXC"</span> <span class="ot">=</span> <span class="st">"1"</span>, <span class="st">"HFJM7BGXC"</span> <span class="ot">=</span> <span class="st">"2"</span>)</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>cmds <span class="ot">&lt;-</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"awk '$0 ~ /Total Sequences/ {{print $0}}' {fastqdata}"</span>)</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cmds) <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(cmds, <span class="st">"(H2HY7BGXC|HFJM7BGXC).+(S</span><span class="sc">\\</span><span class="st">d{2})"</span>)</span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(cmds, \(x) <span class="fu">fread</span>(<span class="at">cmd =</span> x) <span class="sc">%&gt;%</span> .[, V2]) <span class="sc">%&gt;%</span> unlist <span class="sc">%&gt;%</span> enframe <span class="sc">%&gt;%</span> </span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract</span>(name, <span class="fu">c</span>(<span class="st">"batch"</span>, <span class="st">"sampleid"</span>), <span class="st">"(H2HY7BGXC|HFJM7BGXC).+(S</span><span class="sc">\\</span><span class="st">d{2})"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="fu">map_chr</span>(batch, \(x) batches[x])) <span class="sc">%&gt;%</span> </span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sampleid, batch) <span class="sc">%&gt;%</span> </span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">readsbysample =</span> <span class="fu">sum</span>(value), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(ss[library <span class="sc">==</span> <span class="st">"rna-seq"</span>, .(library, sampleid, condition, <span class="at">batch=</span><span class="fu">as.character</span>(batch))] <span class="sc">%&gt;%</span> unique, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sampleid"</span>, <span class="st">"batch"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library, condition, sampleid,  batch, readsbysample) <span class="sc">%&gt;%</span> </span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(readsbysample = scales::comma(readsbysample)) %&gt;% </span></span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(condition,  sampleid, batch) <span class="sc">%&gt;%</span> </span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">format =</span> <span class="st">"html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> library </th>
   <th style="text-align:left;"> condition </th>
   <th style="text-align:left;"> sampleid </th>
   <th style="text-align:left;"> batch </th>
   <th style="text-align:right;"> readsbysample </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-d4 </td>
   <td style="text-align:left;"> S09 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 9698544 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-d4 </td>
   <td style="text-align:left;"> S09 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 12293462 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-d4 </td>
   <td style="text-align:left;"> S10 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 4540513 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-d4 </td>
   <td style="text-align:left;"> S10 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 7162386 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-es </td>
   <td style="text-align:left;"> S03 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7243079 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-es </td>
   <td style="text-align:left;"> S03 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 6271747 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-es </td>
   <td style="text-align:left;"> S04 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 6487837 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k108-es </td>
   <td style="text-align:left;"> S04 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 10041854 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-d4 </td>
   <td style="text-align:left;"> S11 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7408825 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-d4 </td>
   <td style="text-align:left;"> S11 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 11139959 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-d4 </td>
   <td style="text-align:left;"> S12 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8730021 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-d4 </td>
   <td style="text-align:left;"> S12 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 10603993 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-es </td>
   <td style="text-align:left;"> S05 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 9221893 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-es </td>
   <td style="text-align:left;"> S05 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 11072457 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-es </td>
   <td style="text-align:left;"> S06 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 9197964 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> k112-es </td>
   <td style="text-align:left;"> S06 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 19389122 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d1 </td>
   <td style="text-align:left;"> S17 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7553238 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d1 </td>
   <td style="text-align:left;"> S17 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 12884149 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d1 </td>
   <td style="text-align:left;"> S18 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7665723 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d1 </td>
   <td style="text-align:left;"> S18 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 11834691 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d2 </td>
   <td style="text-align:left;"> S19 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8132358 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d2 </td>
   <td style="text-align:left;"> S19 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 8524946 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d2 </td>
   <td style="text-align:left;"> S20 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8296923 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d2 </td>
   <td style="text-align:left;"> S20 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 11276232 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d4 </td>
   <td style="text-align:left;"> S07 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7722012 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d4 </td>
   <td style="text-align:left;"> S07 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 2576710 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d4 </td>
   <td style="text-align:left;"> S08 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 11376879 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d4 </td>
   <td style="text-align:left;"> S08 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 11527405 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d7 </td>
   <td style="text-align:left;"> S21 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7097874 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d7 </td>
   <td style="text-align:left;"> S21 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 8477857 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d7 </td>
   <td style="text-align:left;"> S22 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7327925 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-d7 </td>
   <td style="text-align:left;"> S22 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 11143121 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S01 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8389636 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S01 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 7861728 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S02 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8193903 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S02 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 10278302 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S13 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 9068995 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S13 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 10295966 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S14 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 7775055 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-es </td>
   <td style="text-align:left;"> S14 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 10244819 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-h15 </td>
   <td style="text-align:left;"> S15 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8979579 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-h15 </td>
   <td style="text-align:left;"> S15 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 10030662 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-h15 </td>
   <td style="text-align:left;"> S16 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:right;"> 8459794 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> rna-seq </td>
   <td style="text-align:left;"> wt-h15 </td>
   <td style="text-align:left;"> S16 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:right;"> 9847128 </td>
  </tr>
</tbody>
</table>

</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>