<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chao Dai" />

<meta name="date" content="2022-07-21" />

<title>Encode ChIP-seq Samples</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NeuronReporters</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Encode ChIP-seq Samples</h1>
<h4 class="author">Chao Dai</h4>
<h4 class="date">2022-07-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NeuronReporters/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdaiucNeuronReporterstreee3e83638fb263c5d01da094913e5749799d2ce9atargetblanke3e8363a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/daiuc/NeuronReporters/tree/e3e83638fb263c5d01da094913e5749799d2ce9a" target="_blank">e3e8363</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomdaiucNeuronReporterstreee3e83638fb263c5d01da094913e5749799d2ce9atargetblanke3e8363a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/daiuc/NeuronReporters/tree/e3e83638fb263c5d01da094913e5749799d2ce9a" target="_blank">e3e8363</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.snakemake/
    Ignored:    code/.test/
    Ignored:    code/chip_snakemake.dag
    Ignored:    code/logs/
    Ignored:    code/resources/
    Ignored:    code/results/
    Ignored:    code/slurm_logs/
    Ignored:    code/snakemake_dag.svg
    Ignored:    code/workflow/notebooks/.ipynb_checkpoints/
    Ignored:    code/workflow/notebooks/oldTFscreen/.ipynb_checkpoints/

Untracked files:
    Untracked:  analysis/QC.Rmd
    Untracked:  code/workflow/wf-5-crispr.smk
    Untracked:  code/workflow/wf-6-qc.smk

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   code/workflow/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/EncodeChIPSamples.Rmd</code>) and
HTML (<code>docs/EncodeChIPSamples.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/ae8d00b014e230a213ed64ee29ad533ba31c2608/analysis/EncodeChIPSamples.Rmd" target="_blank">ae8d00b</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
republish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/ae8d00b014e230a213ed64ee29ad533ba31c2608/docs/EncodeChIPSamples.html" target="_blank">ae8d00b</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
republish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/66e0a9c2aa4e5e2302c448383a44aaab4a479822/docs/EncodeChIPSamples.html" target="_blank">66e0a9c</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/4883312939c3e4bbff79727b81ba62b3f3124139/analysis/EncodeChIPSamples.Rmd" target="_blank">4883312</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
wflow_publish("EncodeChIPSamples.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/57be22cb59df71e92737e4e826ef6e0f11433341/docs/EncodeChIPSamples.html" target="_blank">57be22c</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/4b0c0f73c8ba2215b684cf6742e3c155fefc7f49/analysis/EncodeChIPSamples.Rmd" target="_blank">4b0c0f7</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
wflow_publish("EncodeChIPSamples.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/3719c53e2a85ae8f991b9ad628f49e3924df071c/docs/EncodeChIPSamples.html" target="_blank">3719c53</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/252904b9026ff235b0526b51fbe67da8404d5def/analysis/EncodeChIPSamples.Rmd" target="_blank">252904b</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-22
</td>
<td>
wflow_publish("EncodeChIPSamples.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/ea3adfe3af1435db3d2d3c0e8538abd6df3f8fd8/analysis/EncodeChIPSamples.Rmd" target="_blank">ea3adfe</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/ea3adfe3af1435db3d2d3c0e8538abd6df3f8fd8/docs/EncodeChIPSamples.html" target="_blank">ea3adfe</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/f1e20160e26bbc61799a968a96db45af22d255fc/analysis/EncodeChIPSamples.Rmd" target="_blank">f1e2016</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/f1e20160e26bbc61799a968a96db45af22d255fc/docs/EncodeChIPSamples.html" target="_blank">f1e2016</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/d189605e22010a38f83c18916007cdf35627ebed/analysis/EncodeChIPSamples.Rmd" target="_blank">d189605</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/d189605e22010a38f83c18916007cdf35627ebed/docs/EncodeChIPSamples.html" target="_blank">d189605</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/NeuronReporters/blob/fceee7d951deb2be3e2ea5f75929d6f5b69a9279/analysis/EncodeChIPSamples.Rmd" target="_blank">fceee7d</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/NeuronReporters/fceee7d951deb2be3e2ea5f75929d6f5b69a9279/docs/EncodeChIPSamples.html" target="_blank">fceee7d</a>
</td>
<td>
daiuc
</td>
<td>
2022-07-21
</td>
<td>
first commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="encode-tf-chip-seq-experiments" class="section level2">
<h2>Encode TF ChIP-seq experiments</h2>
<p>All human tissue/cell lines:</p>
<pre class="r"><code>encode.meta %&gt;% filter(Organism %in% c(&quot;Homo sapiens&quot;)) %&gt;% 
  select(Organism, Biosample_term_name, Target_gene_symbol, Description) %&gt;% 
  select(Biosample_term_name) %&gt;% unique</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Biosample_term_name"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"HCT116"},{"1":"A549"},{"1":"foreskin fibroblast"},{"1":"H1"},{"1":"retinal pigment epithelial cell"},{"1":"HeLa-S3"},{"1":"K562"},{"1":"GM12878"},{"1":"Karpas-422"},{"1":"Ishikawa"},{"1":"cardiac muscle cell"},{"1":"MCF 10A"},{"1":"NB4"},{"1":"parathyroid adenoma"},{"1":"SK-N-SH"},{"1":"MCF-7"},{"1":"HepG2"},{"1":"WTC11"},{"1":"BLaER1"},{"1":"PC-9"},{"1":"endothelial cell of umbilical vein"},{"1":"IMR-90"},{"1":"myotube"},{"1":"bipolar neuron"},{"1":"Panc1"},{"1":"GM13977"},{"1":"osteoblast"},{"1":"mammary epithelial cell"},{"1":"GM10248"},{"1":"GM12891"},{"1":"GM12801"},{"1":"lung"},{"1":"keratinocyte"},{"1":"OCI-LY7"},{"1":"GM12874"},{"1":"GM12864"},{"1":"T47D"},{"1":"fibroblast of lung"},{"1":"kidney"},{"1":"AG08470"},{"1":"GM19099"},{"1":"WI38"},{"1":"GM18505"},{"1":"HFF-Myc"},{"1":"GM15510"},{"1":"HEK293"},{"1":"AG10803"},{"1":"lower leg skin"},{"1":"OCI-LY3"},{"1":"HEK293T"},{"1":"PC-3"},{"1":"prostate gland"},{"1":"liver"},{"1":"GM23338"},{"1":"transverse colon"},{"1":"omental fat pad"},{"1":"neural progenitor cell"},{"1":"stomach"},{"1":"neutrophil"},{"1":"GM19239"},{"1":"body of pancreas"},{"1":"gastroesophageal sphincter"},{"1":"Caco-2"},{"1":"GM10847"},{"1":"HL-60"},{"1":"hepatocyte"},{"1":"epithelial cell of esophagus"},{"1":"subcutaneous adipose tissue"},{"1":"GM12868"},{"1":"GM12869"},{"1":"foreskin keratinocyte"},{"1":"RWPE2"},{"1":"upper lobe of left lung"},{"1":"GM12872"},{"1":"DOHH2"},{"1":"choroid plexus epithelial cell"},{"1":"HGPS cell"},{"1":"skeletal muscle myoblast"},{"1":"fibroblast of dermis"},{"1":"fibroblast of pulmonary artery"},{"1":"SU-DHL-6"},{"1":"suprapubic skin"},{"1":"GM12892"},{"1":"SK-N-MC"},{"1":"sigmoid colon"},{"1":"BJ"},{"1":"GM19240"},{"1":"breast epithelium"},{"1":"SH-SY5Y"},{"1":"GM12875"},{"1":"KMS-11"},{"1":"erythroblast"},{"1":"GM12873"},{"1":"spleen"},{"1":"Loucy"},{"1":"neural cell"},{"1":"gastrocnemius medialis"},{"1":"U2OS"},{"1":"GM19193"},{"1":"BE2C"},{"1":"thyroid gland"},{"1":"AG04449"},{"1":"bronchial epithelial cell"},{"1":"GM08714"},{"1":"tibial nerve"},{"1":"adrenal gland"},{"1":"PFSK-1"},{"1":"vagina"},{"1":"astrocyte"},{"1":"epithelial cell of proximal tubule"},{"1":"B cell"},{"1":"GM20000"},{"1":"cardiac fibroblast"},{"1":"LNCAP"},{"1":"C4-2B"},{"1":"GM10266"},{"1":"CD14-positive monocyte"},{"1":"testis"},{"1":"LNCaP clone FGC"},{"1":"NT2/D1"},{"1":"22Rv1"},{"1":"U-87 MG"},{"1":"GM12867"},{"1":"GM12871"},{"1":"OCI-LY1"},{"1":"GM12870"},{"1":"Raji"},{"1":"AG04450"},{"1":"D721Med"},{"1":"astrocyte of the spinal cord"},{"1":"GM18526"},{"1":"fibroblast of villous mesenchyme"},{"1":"GM18951"},{"1":"GM19238"},{"1":"GM12866"},{"1":"fibroblast of mammary gland"},{"1":"brain microvascular endothelial cell"},{"1":"astrocyte of the cerebellum"},{"1":"fibroblast of the aortic adventitia"},{"1":"DND-41"},{"1":"AG09319"},{"1":"GM06990"},{"1":"kidney epithelial cell"},{"1":"GM12865"},{"1":"NCI-H929"},{"1":"A673"},{"1":"pancreas"},{"1":"AG09309"},{"1":"Peyer's patch"},{"1":"GM23248"},{"1":"right lobe of liver"},{"1":"H54"},{"1":"esophagus squamous epithelium"},{"1":"WERI-Rb-1"},{"1":"GM13976"},{"1":"MM.1S"},{"1":"HFFc6"},{"1":"middle frontal area 46"},{"1":"left lung"},{"1":"brain"},{"1":"VCaP"},{"1":"esophagus muscularis mucosa"},{"1":"smooth muscle cell"},{"1":"ovary"},{"1":"uterus"},{"1":"activated CD4-positive, alpha-beta T cell"},{"1":"osteocyte"},{"1":"lower lobe of right lung"},{"1":"heart right ventricle"},{"1":"upper lobe of right lung"},{"1":"heart left ventricle"},{"1":"thoracic aorta"},{"1":"epithelial cell of prostate"},{"1":"coronary artery"},{"1":"right atrium auricular region"},{"1":"RWPE1"},{"1":"tibial artery"},{"1":"ascending aorta"},{"1":"H9"},{"1":"colonic mucosa"},{"1":"type B pancreatic cell"},{"1":"CD8-positive, alpha-beta T cell"},{"1":"placenta"},{"1":"activated B cell"},{"1":"excitatory neuron"},{"1":"mesothelial cell of epicardium"},{"1":"Calu3"},{"1":"endothelial cell"},{"1":"left ventricle myocardium inferior"},{"1":"endodermal cell"},{"1":"mucosa of descending colon"},{"1":"chondrocyte"},{"1":"neural crest cell"},{"1":"lower lobe of left lung"},{"1":"nephron"},{"1":"CD4-positive, alpha-beta T cell"},{"1":"progenitor cell of endocrine pancreas"},{"1":"psoas muscle"},{"1":"nephron progenitor cell"},{"1":"natural killer cell"},{"1":"activated CD8-positive, alpha-beta T cell"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>From the above table, I selected these tissue types as relevant:</p>
<pre><code>&quot;bipolar neuron&quot;, &quot;neural progenitor cell&quot;, &quot;neural cell&quot;, &quot;astrocyte of the cerebellum&quot;,
&quot;brain&quot;, &quot;excitatory neuron&quot;, &quot;neural crest cell&quot;</code></pre>
<pre class="r"><code>neuron_related_tissues = c(&quot;bipolar neuron&quot;, &quot;neural progenitor cell&quot;, &quot;neural cell&quot;, 
                           &quot;astrocyte of the cerebellum&quot;,&quot;brain&quot;, &quot;excitatory neuron&quot;, 
                           &quot;neural crest cell&quot;)</code></pre>
<p>ChIP-seq on the following gene products are available:</p>
<pre class="r"><code>encode.meta %&gt;% 
  select(Organism, Biosample_term_name, Target_gene_symbol, Description) %&gt;% 
  filter(Organism %in% c(&quot;Homo sapiens&quot;) &amp; Biosample_term_name %in% neuron_related_tissues)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Organism"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Biosample_term_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Target_gene_symbol"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Description"],"name":[4],"type":["chr"],"align":["left"]}],"data":[{"1":"Homo sapiens","2":"bipolar neuron","3":"CTCF","4":""},{"1":"Homo sapiens","2":"neural progenitor cell","3":"EZH2","4":""},{"1":"Homo sapiens","2":"bipolar neuron","3":"ZEB1","4":"Chip-Seq on biopolar spindle neuron"},{"1":"Homo sapiens","2":"bipolar neuron","3":"POU5F1","4":"Chip-Seq on biopolar spindle neuron"},{"1":"Homo sapiens","2":"neural progenitor cell","3":"CTCF","4":""},{"1":"Homo sapiens","2":"neural cell","3":"SMC3","4":"SMC3 ChIP-seq on human neural cell"},{"1":"Homo sapiens","2":"neural cell","3":"MXI1","4":"MXI1 ChIP-seq on human neural cell"},{"1":"Homo sapiens","2":"neural cell","3":"RAD21","4":"RAD21 ChIP-seq on human neural cell"},{"1":"Homo sapiens","2":"neural progenitor cell","3":"EZH2","4":""},{"1":"Homo sapiens","2":"bipolar neuron","3":"EZH2","4":""},{"1":"Homo sapiens","2":"neural cell","3":"POLR2A","4":"POLR2AphosphoS5 ChIP-seq protocol v042211.1 on human H1-neurons"},{"1":"Homo sapiens","2":"neural cell","3":"REST","4":"REST ChIP-seq protocol v042211.1 on human H1-neurons"},{"1":"Homo sapiens","2":"neural cell","3":"EZH2","4":""},{"1":"Homo sapiens","2":"astrocyte of the cerebellum","3":"CTCF","4":"CTCF ChIP-seq on human HAc"},{"1":"Homo sapiens","2":"neural cell","3":"TAF1","4":"TAF1 ChIP-seq protocol v042211.1 on human H1-neurons"},{"1":"Homo sapiens","2":"neural cell","3":"CTCF","4":"CTCF ChIP-seq on human neural cell"},{"1":"Homo sapiens","2":"neural cell","3":"EP300","4":"EP300 ChIP-seq on human neural cell"},{"1":"Homo sapiens","2":"brain","3":"CTCF","4":"CTCF ChIP-seq on brain female embryo (5 days)"},{"1":"Homo sapiens","2":"neural progenitor cell","3":"CTCF","4":"CTCF ChIP-seq on neural progenitor cell"},{"1":"Homo sapiens","2":"brain","3":"CTCF","4":"CTCF ChIP-seq on brain female embryo (5 days)"},{"1":"Homo sapiens","2":"bipolar neuron","3":"SMARCA4","4":"Chip-Seq on biopolar spindle neuron"},{"1":"Homo sapiens","2":"bipolar neuron","3":"POLR2A","4":"Chip-Seq on biopolar spindle neuron"},{"1":"Homo sapiens","2":"bipolar neuron","3":"CTCF","4":"Chip-Seq on biopolar spindle neuron"},{"1":"Homo sapiens","2":"brain","3":"CTCF","4":""},{"1":"Homo sapiens","2":"excitatory neuron","3":"CTCF","4":"CTCF ChIP-seq on excitatory neuron"},{"1":"Homo sapiens","2":"neural crest cell","3":"CTCF","4":"CTCF ChIP-seq on neural crest cell originated from neural crest cell"},{"1":"Homo sapiens","2":"brain","3":"CTCF","4":"CTCF ChIP-seq on brain female embryo (5 days)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>chip.avail.genes = filter(encode.meta, 
                          Organism %in% c(&quot;Homo sapiens&quot;) &amp; 
                            Biosample_term_name %in% neuron_related_tissues) %&gt;% 
  pull(Target_gene_symbol) %&gt;% unique

chip.avail.genes</code></pre>
<pre><code> [1] &quot;CTCF&quot;    &quot;EZH2&quot;    &quot;ZEB1&quot;    &quot;POU5F1&quot;  &quot;SMC3&quot;    &quot;MXI1&quot;    &quot;RAD21&quot;  
 [8] &quot;POLR2A&quot;  &quot;REST&quot;    &quot;TAF1&quot;    &quot;EP300&quot;   &quot;SMARCA4&quot;</code></pre>
</div>
<div id="interaction-matrix-from-neuron-reporters"
class="section level2">
<h2>Interaction matrix from Neuron Reporters</h2>
<pre class="r"><code>my.inter = &quot;../code/resources/Interaction_matrix_nofilter_AllTF_20200107.csv&quot;
my.inter = fread(my.inter)</code></pre>
<p>There are 456 unique TFs recognized as regulators in our network
analyses, shown in table below:</p>
<pre class="r"><code>select(my.inter, regulator_gene) %&gt;% unique</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["regulator_gene"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"AR"},{"1":"ARNT2"},{"1":"ATF2"},{"1":"ATF3"},{"1":"ATF4"},{"1":"ATF6"},{"1":"ATF7"},{"1":"ATOH1"},{"1":"BACH1"},{"1":"BACH2"},{"1":"BARHL2"},{"1":"BARX1"},{"1":"BARX2"},{"1":"BATF"},{"1":"BATF3"},{"1":"BCL6"},{"1":"BCL6B"},{"1":"BHLHE40"},{"1":"BHLHE41"},{"1":"CDX1"},{"1":"CDX4"},{"1":"CEBPA"},{"1":"CEBPB"},{"1":"CEBPD"},{"1":"CEBPG"},{"1":"CENPB"},{"1":"CLOCK"},{"1":"CREB1"},{"1":"CREB3"},{"1":"CREB3L1"},{"1":"CREB3L4"},{"1":"CREM"},{"1":"CTCF"},{"1":"CTCFL"},{"1":"CUX1"},{"1":"CUX2"},{"1":"DBP"},{"1":"DLX6"},{"1":"DMRT3"},{"1":"DMRTC2"},{"1":"E2F1"},{"1":"E2F2"},{"1":"E2F3"},{"1":"E2F4"},{"1":"E2F6"},{"1":"E2F8"},{"1":"EBF1"},{"1":"EBF3"},{"1":"EGR1"},{"1":"EGR2"},{"1":"EGR3"},{"1":"EGR4"},{"1":"ELF1"},{"1":"ELF2"},{"1":"ELF3"},{"1":"ELF4"},{"1":"ELF5"},{"1":"ELK1"},{"1":"ELK3"},{"1":"ELK4"},{"1":"EMX1"},{"1":"EMX2"},{"1":"EN1"},{"1":"EN2"},{"1":"EOMES"},{"1":"ERF"},{"1":"ERG"},{"1":"ESR1"},{"1":"ESR2"},{"1":"ESRRA"},{"1":"ESRRB"},{"1":"ETS1"},{"1":"ETS2"},{"1":"ETV1"},{"1":"ETV2"},{"1":"ETV3"},{"1":"ETV4"},{"1":"ETV5"},{"1":"ETV6"},{"1":"EVX1"},{"1":"FEV"},{"1":"FIGLA"},{"1":"FLI1"},{"1":"FOS"},{"1":"FOSL1"},{"1":"FOSL2"},{"1":"FOXA1"},{"1":"FOXA2"},{"1":"FOXA3"},{"1":"FOXB1"},{"1":"FOXC1"},{"1":"FOXC2"},{"1":"FOXD2"},{"1":"FOXE1"},{"1":"FOXF2"},{"1":"FOXG1"},{"1":"FOXH1"},{"1":"FOXK2"},{"1":"FOXL1"},{"1":"FOXN3"},{"1":"FOXO3"},{"1":"FOXO4"},{"1":"FOXP1"},{"1":"FOXP2"},{"1":"FOXP3"},{"1":"GABPA"},{"1":"GATA2"},{"1":"GATA3"},{"1":"GATA4"},{"1":"GATA5"},{"1":"GATA6"},{"1":"GBX2"},{"1":"GCM1"},{"1":"GCM2"},{"1":"GFI1"},{"1":"GLI2"},{"1":"GLI3"},{"1":"GLIS1"},{"1":"GLIS2"},{"1":"GLIS3"},{"1":"GMEB2"},{"1":"GRHL1"},{"1":"GSC"},{"1":"HAND2"},{"1":"HES1"},{"1":"HES2"},{"1":"HES6"},{"1":"HESX1"},{"1":"HEY1"},{"1":"HEY2"},{"1":"HIC2"},{"1":"HIF1A"},{"1":"HINFP"},{"1":"HLF"},{"1":"HMBOX1"},{"1":"HNF1A"},{"1":"HNF1B"},{"1":"HNF4A"},{"1":"HNF4G"},{"1":"HOXA1"},{"1":"HOXA13"},{"1":"HOXA5"},{"1":"HOXA6"},{"1":"HOXA7"},{"1":"HOXB2"},{"1":"HOXB3"},{"1":"HOXB4"},{"1":"HOXB7"},{"1":"HOXB8"},{"1":"HOXC11"},{"1":"HOXC13"},{"1":"HOXC8"},{"1":"HOXD13"},{"1":"HOXD3"},{"1":"HOXD8"},{"1":"HOXD9"},{"1":"HSF1"},{"1":"HSF2"},{"1":"HSF4"},{"1":"IKZF1"},{"1":"INSM1"},{"1":"IRF1"},{"1":"IRF2"},{"1":"IRF3"},{"1":"IRF4"},{"1":"IRF5"},{"1":"IRF6"},{"1":"IRF7"},{"1":"IRF8"},{"1":"ISL2"},{"1":"JDP2"},{"1":"JUN"},{"1":"JUNB"},{"1":"JUND"},{"1":"KLF10"},{"1":"KLF11"},{"1":"KLF13"},{"1":"KLF15"},{"1":"KLF16"},{"1":"KLF2"},{"1":"KLF3"},{"1":"KLF4"},{"1":"KLF5"},{"1":"KLF6"},{"1":"KLF9"},{"1":"LEF1"},{"1":"LHX1"},{"1":"LHX2"},{"1":"LHX5"},{"1":"LHX6"},{"1":"LHX9"},{"1":"LMX1B"},{"1":"MAF"},{"1":"MAFA"},{"1":"MAFF"},{"1":"MAFG"},{"1":"MAFK"},{"1":"MAX"},{"1":"MAZ"},{"1":"MEF2A"},{"1":"MEF2C"},{"1":"MEF2D"},{"1":"MEIS1"},{"1":"MEIS2"},{"1":"MEIS3"},{"1":"MEOX1"},{"1":"MEOX2"},{"1":"MITF"},{"1":"MIXL1"},{"1":"MLX"},{"1":"MLXIPL"},{"1":"MNT"},{"1":"MNX1"},{"1":"MSC"},{"1":"MSX1"},{"1":"MSX2"},{"1":"MTF1"},{"1":"MXI1"},{"1":"MYB"},{"1":"MYBL1"},{"1":"MYBL2"},{"1":"MYC"},{"1":"MYCN"},{"1":"MYF5"},{"1":"MZF1"},{"1":"NEUROD1"},{"1":"NEUROD2"},{"1":"NEUROG1"},{"1":"NEUROG2"},{"1":"NFATC2"},{"1":"NFATC3"},{"1":"NFATC4"},{"1":"NFE2"},{"1":"NFE2L1"},{"1":"NFE2L2"},{"1":"NFIA"},{"1":"NFIB"},{"1":"NFIC"},{"1":"NFIL3"},{"1":"NFIX"},{"1":"NFKB1"},{"1":"NFKB2"},{"1":"NFYA"},{"1":"NFYB"},{"1":"NFYC"},{"1":"NHLH1"},{"1":"NHLH2"},{"1":"NKX2-2"},{"1":"NKX2-5"},{"1":"NKX2-8"},{"1":"NKX3-1"},{"1":"NKX3-2"},{"1":"NKX6-1"},{"1":"NKX6-2"},{"1":"NR1D1"},{"1":"NR1D2"},{"1":"NR1I2"},{"1":"NR1I3"},{"1":"NR2C1"},{"1":"NR2C2"},{"1":"NR2F1"},{"1":"NR2F2"},{"1":"NR2F6"},{"1":"NR3C1"},{"1":"NR3C2"},{"1":"NR4A1"},{"1":"NR4A2"},{"1":"NRF1"},{"1":"NRL"},{"1":"OLIG1"},{"1":"OLIG2"},{"1":"OLIG3"},{"1":"ONECUT1"},{"1":"ONECUT2"},{"1":"OTX1"},{"1":"OTX2"},{"1":"OVOL1"},{"1":"OVOL2"},{"1":"PAX1"},{"1":"PAX3"},{"1":"PAX4"},{"1":"PAX5"},{"1":"PAX6"},{"1":"PAX9"},{"1":"PBX1"},{"1":"PBX2"},{"1":"PBX3"},{"1":"PHOX2A"},{"1":"PHOX2B"},{"1":"PITX1"},{"1":"PITX2"},{"1":"PITX3"},{"1":"PKNOX1"},{"1":"PKNOX2"},{"1":"PLAG1"},{"1":"PLAGL2"},{"1":"POU2F1"},{"1":"POU2F2"},{"1":"POU2F3"},{"1":"POU3F1"},{"1":"POU3F2"},{"1":"POU3F3"},{"1":"POU3F4"},{"1":"POU4F1"},{"1":"POU4F2"},{"1":"POU4F3"},{"1":"POU5F1"},{"1":"POU5F1B"},{"1":"POU6F1"},{"1":"POU6F2"},{"1":"PPARD"},{"1":"PPARG"},{"1":"PRDM1"},{"1":"PRDM4"},{"1":"PROX1"},{"1":"PRRX1"},{"1":"PRRX2"},{"1":"RARA"},{"1":"RARB"},{"1":"RARG"},{"1":"RAX"},{"1":"RBPJ"},{"1":"REL"},{"1":"RELA"},{"1":"RELB"},{"1":"REST"},{"1":"RFX1"},{"1":"RFX2"},{"1":"RFX3"},{"1":"RFX4"},{"1":"RFX5"},{"1":"RHOXF1"},{"1":"RORA"},{"1":"RORB"},{"1":"RORC"},{"1":"RREB1"},{"1":"RUNX1"},{"1":"RUNX2"},{"1":"RUNX3"},{"1":"RXRB"},{"1":"RXRG"},{"1":"SCRT1"},{"1":"SCRT2"},{"1":"SHOX"},{"1":"SIX1"},{"1":"SIX2"},{"1":"SMAD3"},{"1":"SNAI1"},{"1":"SNAI2"},{"1":"SNAI3"},{"1":"SOX10"},{"1":"SOX12"},{"1":"SOX13"},{"1":"SOX14"},{"1":"SOX15"},{"1":"SOX2"},{"1":"SOX21"},{"1":"SOX4"},{"1":"SOX8"},{"1":"SOX9"},{"1":"SP1"},{"1":"SP2"},{"1":"SP3"},{"1":"SP4"},{"1":"SP8"},{"1":"SPDEF"},{"1":"SREBF1"},{"1":"SREBF2"},{"1":"SRF"},{"1":"STAT1"},{"1":"STAT3"},{"1":"TBR1"},{"1":"TBX1"},{"1":"TBX15"},{"1":"TBX18"},{"1":"TBX19"},{"1":"TBX2"},{"1":"TBX3"},{"1":"TBX5"},{"1":"TBX6"},{"1":"TCF12"},{"1":"TCF21"},{"1":"TCF3"},{"1":"TCF4"},{"1":"TCF7"},{"1":"TCF7L1"},{"1":"TCF7L2"},{"1":"TCFL5"},{"1":"TEAD1"},{"1":"TEAD2"},{"1":"TEAD3"},{"1":"TEAD4"},{"1":"TEF"},{"1":"TFAP2A"},{"1":"TFAP2B"},{"1":"TFAP2C"},{"1":"TFAP2E"},{"1":"TFAP4"},{"1":"TFCP2"},{"1":"TFDP1"},{"1":"TFE3"},{"1":"TFEB"},{"1":"TFEC"},{"1":"TGIF1"},{"1":"TGIF2"},{"1":"THRB"},{"1":"TLX2"},{"1":"TP53"},{"1":"TP63"},{"1":"TP73"},{"1":"TWIST1"},{"1":"USF1"},{"1":"USF2"},{"1":"VAX1"},{"1":"VAX2"},{"1":"VDR"},{"1":"VENTX"},{"1":"VEZF1"},{"1":"VSX1"},{"1":"VSX2"},{"1":"XBP1"},{"1":"YY1"},{"1":"ZBTB14"},{"1":"ZBTB18"},{"1":"ZBTB26"},{"1":"ZBTB32"},{"1":"ZBTB33"},{"1":"ZBTB6"},{"1":"ZBTB7A"},{"1":"ZBTB7B"},{"1":"ZEB1"},{"1":"ZFP42"},{"1":"ZFP57"},{"1":"ZIC1"},{"1":"ZIC3"},{"1":"ZKSCAN1"},{"1":"ZKSCAN5"},{"1":"ZNF135"},{"1":"ZNF136"},{"1":"ZNF140"},{"1":"ZNF143"},{"1":"ZNF148"},{"1":"ZNF16"},{"1":"ZNF24"},{"1":"ZNF263"},{"1":"ZNF282"},{"1":"ZNF317"},{"1":"ZNF341"},{"1":"ZNF354C"},{"1":"ZNF382"},{"1":"ZNF384"},{"1":"ZNF410"},{"1":"ZNF449"},{"1":"ZNF460"},{"1":"ZNF75D"},{"1":"ZSCAN4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Found only regulators having ChIP-seq data.</p>
<pre class="r"><code>intersect(my.inter$regulator_gene, chip.avail.genes)</code></pre>
<pre><code>[1] &quot;CTCF&quot;   &quot;MXI1&quot;   &quot;POU5F1&quot; &quot;REST&quot;   &quot;ZEB1&quot;  </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /gpfs/commons/home/cdai/miniconda3/envs/smk/lib/libopenblasp-r0.3.18.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.2 forcats_0.5.1     stringr_1.4.0     dplyr_1.0.9      
 [5] purrr_0.3.4       readr_2.1.2       tidyr_1.2.0       tibble_3.1.7     
 [9] ggplot2_3.3.6     tidyverse_1.3.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9          lubridate_1.8.0     assertthat_0.2.1   
 [4] rprojroot_2.0.3     digest_0.6.29       utf8_1.2.2         
 [7] R6_2.5.1            cellranger_1.1.0    backports_1.4.1    
[10] reprex_2.0.1        evaluate_0.15       httr_1.4.3         
[13] pillar_1.8.0        rlang_1.0.4         googlesheets4_1.0.0
[16] readxl_1.4.0        rstudioapi_0.13     whisker_0.4        
[19] jquerylib_0.1.4     rmarkdown_2.14      googledrive_2.0.0  
[22] munsell_0.5.0       broom_1.0.0         compiler_4.1.0     
[25] httpuv_1.6.5        modelr_0.1.8        xfun_0.31          
[28] pkgconfig_2.0.3     htmltools_0.5.3     tidyselect_1.1.2   
[31] workflowr_1.7.0     fansi_1.0.3         crayon_1.5.1       
[34] withr_2.5.0         tzdb_0.3.0          dbplyr_2.2.1       
[37] later_1.3.0         grid_4.1.0          jsonlite_1.8.0     
[40] gtable_0.3.0        lifecycle_1.0.1     DBI_1.1.3          
[43] git2r_0.30.1        magrittr_2.0.3      scales_1.2.0       
[46] cli_3.3.0           stringi_1.7.8       cachem_1.0.6       
[49] fs_1.5.2            promises_1.2.0.1    xml2_1.3.3         
[52] bslib_0.4.0         ellipsis_0.3.2      generics_0.1.3     
[55] vctrs_0.4.1         tools_4.1.0         glue_1.6.2         
[58] hms_1.1.1           fastmap_1.1.0       yaml_2.3.5         
[61] colorspace_2.0-3    gargle_1.2.0        rvest_1.0.2        
[64] knitr_1.39          haven_2.5.0         sass_0.4.2         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
