<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>MAGeCK results</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NeuronReporters</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MAGeCK results</h1>

</div>


<pre class="r"><code>knitr::opts_chunk$set(include = T, warning = F, message = F)

library(tidyverse)
library(data.table)
library(EnhancedVolcano)


theme_set(theme_bw())
cnts = fread(&quot;../../resources/crispr/counts/wrangled_raw_counts.tsv&quot;)</code></pre>
<div id="rra" class="section level2">
<h2>RRA</h2>
<p>Run RRA in 3 contrasts:</p>
<ul>
<li>Day14_Tdneg-vs-Day14_iN<br />
</li>
<li>Day14_Tdneg-vs-Day14_Tdpos<br />
</li>
<li>Day14_Tdpos-vs-Day14_iN</li>
</ul>
<hr />
<div id="day14_tdneg-vs-day14_in---look-for-neuron-differentiation-activators" class="section level3">
<h3>Day14_Tdneg-vs-Day14_iN - look for neuron differentiation activators</h3>
<p>For Tdtomato - vs. iN, we look for enriched genes with guides that are lost. TFs with significant reduction in sgRNA reads in the tdTomato- cells indicate that these TFs can potentially positively regulate neuron differentiation.</p>
<pre class="r"><code># read in data
tdneg.vs.iN = fread(&quot;../../results/crispr/mageckRRA/Day14_Tdneg-vs-Day14_iN/RRA.gene_summary.txt&quot;)</code></pre>
<pre class="r"><code>tdneg.vs.iN[, .(id, `neg|score`, `neg|lfc`)] %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% 
  EnhancedVolcano(., rownames(.), x = &quot;neg|lfc&quot;, y = &quot;neg|score&quot;, 
                  title = NULL, subtitle = NULL, caption = NULL,
                  pCutoff = 1e-3, FCcutoff = 1, max.overlaps = 30,
                  labSize = 3, legendLabSize = 10, legendIconSize = 3,
                  ylim = c(0, 6)
                  ) +
  theme(axis.title = element_text(size = 10),
        axis.text.x = element_text(size = 10), axis.text.y = element_text(size = 10),
        legend.position = &quot;None&quot;
        )</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
</div>
<div id="day14_tdneg-vs-day14_tdpos---look-for-neuron-differentiation-activators" class="section level3">
<h3>Day14_Tdneg-vs-Day14_Tdpos - look for neuron differentiation activators</h3>
<p>For Tdtomato- vs . Tdtomato+ we look for genes with guides that are lost. Similarly to above, TFs with significantly reduced sgRNA reads in the tdTomato- cells vs. tdTomato+ cells may indicate that these TFs can promote neuron differentiation.</p>
<pre class="r"><code># read in data
tdneg.vs.tdpos = fread(&quot;../../results/crispr/mageckRRA/Day14_Tdneg-vs-Day14_Tdpos/RRA.gene_summary.txt&quot;)</code></pre>
<pre class="r"><code>tdneg.vs.tdpos[, .(id, `neg|score`, `neg|lfc`)] %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% 
  EnhancedVolcano(., rownames(.), x = &quot;neg|lfc&quot;, y = &quot;neg|score&quot;, 
                  title = NULL, subtitle = NULL, caption = NULL,
                  pCutoff = 1e-3, FCcutoff = 1, max.overlaps = 30,
                  labSize = 3, legendLabSize = 10, legendIconSize = 3,
                  ylim = c(0, 6)
                  ) +
  theme(axis.title = element_text(size = 10),
        axis.text.x = element_text(size = 10), axis.text.y = element_text(size = 10),
        legend.position = &quot;None&quot;
        )</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="day14_tdpos-vs-day14_in---look-for-neuron-differentiation-inhibitors" class="section level3">
<h3>Day14_Tdpos-vs-Day14_iN - look for neuron differentiation inhibitors</h3>
<p>In Tdtomato+ vs iN contrast, we look for TFs with significantly reduced sgRNA reads. These indicates that the knock out of these TFs promoted neuron differentiation, suggesting that these TFs are likely inhibitors of neuron differentiation.</p>
<p><br></p>
<p>Why do we also use negative screen in tdTomato+ vs. iN? Because this is a CRISPR-cas9 knockout screen, in theory, we should only observe targeted sgRNA reads to decrease.</p>
<pre class="r"><code># read in data
tdpos.vs.iN = fread(&quot;../../results/crispr/mageckRRA/Day14_Tdpos-vs-Day14_iN/RRA.gene_summary.txt&quot;)</code></pre>
<pre class="r"><code>tdpos.vs.iN[, .(id, `neg|score`, `neg|lfc`)] %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% 
  EnhancedVolcano(., rownames(.), x = &quot;neg|lfc&quot;, y = &quot;neg|score&quot;, 
                  title = NULL, subtitle = NULL, caption = NULL,
                  pCutoff = 1e-3, FCcutoff = 1, max.overlaps = 30,
                  labSize = 3, legendLabSize = 10, legendIconSize = 3,
                  ylim = c(0, 6)
                  ) +
  theme(axis.title = element_text(size = 10),
        axis.text.x = element_text(size = 10), axis.text.y = element_text(size = 10),
        legend.position = &quot;None&quot;
        )</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
</div>
<div id="compare-rra-scores" class="section level2">
<h2>Compare RRA scores</h2>
<pre class="r"><code>inner_join(x = tdneg.vs.iN[, .(id, tdneg_vs_iN = `neg|score`)],
           y = tdneg.vs.tdpos[, .(id, tdneg_vs_tdpos = `neg|score`)], 
           by = &quot;id&quot;
           ) %&gt;% 
  ggplot() + geom_point(aes(tdneg_vs_iN, tdneg_vs_tdpos), alpha = .3, color = &quot;royalblue&quot;)</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>inner_join(x = tdneg.vs.iN[, .(id, tdneg_vs_iN = `neg|score`)],
           y = tdpos.vs.iN[, .(id, tdpos_vs_iN = `neg|score`)], 
           by = &quot;id&quot;
           ) %&gt;% 
  ggplot() + geom_point(aes(tdneg_vs_iN, tdpos_vs_iN), alpha = .3, color = &quot;firebrick&quot;)</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
<div id="compare-lfc" class="section level2">
<h2>Compare lfc</h2>
<p>Comparison of log(fold change) shows that indeed, tdneg_vs_iN and tdneg_vs_tdpos is highly correlated.</p>
<pre class="r"><code>inner_join(x = tdneg.vs.iN[, .(id, tdneg_vs_iN = `neg|lfc`)],
           y = tdneg.vs.tdpos[, .(id, tdneg_vs_tdpos = `neg|lfc`)], 
           by = &quot;id&quot;
           ) %&gt;% 
  ggplot() + geom_point(aes(tdneg_vs_iN, tdneg_vs_tdpos), alpha = .3, color = &quot;royalblue&quot;)</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre class="r"><code>inner_join(x = tdneg.vs.iN[, .(id, tdneg_vs_iN = `neg|lfc`)],
           y = tdpos.vs.iN[, .(id, tdpos_vs_iN = `neg|lfc`)], 
           by = &quot;id&quot;
           ) %&gt;% 
  ggplot() + geom_point(aes(tdneg_vs_iN, tdpos_vs_iN), alpha = .3, color = &quot;firebrick&quot;)</code></pre>
<p><img src="MAGeCK_results_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
<div id="intersect-tdtomato--vs.-in-and-tdtomato--vs.-tdtomato-top-hits" class="section level2">
<h2>Intersect tdTomato- vs. iN and tdTomato- vs. tdTomato+ top hits</h2>
<p>Intersected TFs with <code>neg|score &lt; 1e-3</code>.</p>
<pre class="r"><code>intersect(tdneg.vs.iN[`neg|score` &lt; 1e-3, id], tdneg.vs.tdpos[`neg|score` &lt; 1e-3, id])</code></pre>
<pre><code>##  [1] &quot;AES&quot;                     &quot;HSFY1&quot;                  
##  [3] &quot;SND1&quot;                    &quot;SOX11&quot;                  
##  [5] &quot;TP53&quot;                    &quot;SOX4&quot;                   
##  [7] &quot;CHD7&quot;                    &quot;ZSCAN23&quot;                
##  [9] &quot;NCOR2&quot;                   &quot;CNOT4&quot;                  
## [11] &quot;NonTargeting_Human_0399&quot; &quot;NonTargeting_Human_0086&quot;
## [13] &quot;MED13&quot;                   &quot;POU4F2&quot;                 
## [15] &quot;TGIF2LY&quot;                 &quot;HMGN5&quot;                  
## [17] &quot;NonTargeting_Human_0381&quot; &quot;NonTargeting_Human_0287&quot;
## [19] &quot;RBM15&quot;                   &quot;ARID3A&quot;                 
## [21] &quot;NonTargeting_Human_0713&quot;</code></pre>
<p>Intersects of top 200 TFs</p>
<pre class="r"><code>intersect(tdneg.vs.iN[`neg|rank` &lt; 201, id], tdneg.vs.tdpos[`neg|rank` &lt; 201, id])</code></pre>
<pre><code>##   [1] &quot;AES&quot;                     &quot;HSFY1&quot;                  
##   [3] &quot;SND1&quot;                    &quot;SOX11&quot;                  
##   [5] &quot;TP53&quot;                    &quot;SOX4&quot;                   
##   [7] &quot;CTNNBIP1&quot;                &quot;CHD7&quot;                   
##   [9] &quot;ZSCAN23&quot;                 &quot;NHLH2&quot;                  
##  [11] &quot;NCOR2&quot;                   &quot;CNOT4&quot;                  
##  [13] &quot;NonTargeting_Human_0399&quot; &quot;PKNOX2&quot;                 
##  [15] &quot;NonTargeting_Human_0086&quot; &quot;MED13&quot;                  
##  [17] &quot;POU4F2&quot;                  &quot;TGIF2LY&quot;                
##  [19] &quot;OLIG2&quot;                   &quot;HMGN5&quot;                  
##  [21] &quot;NonTargeting_Human_0381&quot; &quot;OVOL2&quot;                  
##  [23] &quot;NonTargeting_Human_0287&quot; &quot;HOXC10&quot;                 
##  [25] &quot;RAX&quot;                     &quot;RBM15&quot;                  
##  [27] &quot;NonTargeting_Human_0829&quot; &quot;KDM5D&quot;                  
##  [29] &quot;TCF4&quot;                    &quot;TCF3&quot;                   
##  [31] &quot;ARID3A&quot;                  &quot;NonTargeting_Human_0573&quot;
##  [33] &quot;NonTargeting_Human_0713&quot; &quot;NonTargeting_Human_0778&quot;
##  [35] &quot;HOXD9&quot;                   &quot;ZNF626&quot;                 
##  [37] &quot;MNDA&quot;                    &quot;SUPT7L&quot;                 
##  [39] &quot;NonTargeting_Human_0692&quot; &quot;NonTargeting_Human_0905&quot;
##  [41] &quot;NonTargeting_Human_0785&quot; &quot;NonTargeting_Human_0490&quot;
##  [43] &quot;FOXH1&quot;                   &quot;NonTargeting_Human_0586&quot;
##  [45] &quot;NonTargeting_Human_0999&quot; &quot;TBPL1&quot;                  
##  [47] &quot;FOXO4&quot;                   &quot;NonTargeting_Human_0735&quot;
##  [49] &quot;NonTargeting_Human_0480&quot; &quot;NKX6-1&quot;                 
##  [51] &quot;NonTargeting_Human_0082&quot; &quot;MEN1&quot;                   
##  [53] &quot;NonTargeting_Human_0531&quot; &quot;DPF1&quot;                   
##  [55] &quot;NonTargeting_Human_0916&quot; &quot;NonTargeting_Human_0010&quot;
##  [57] &quot;NHLH1&quot;                   &quot;SOX21&quot;                  
##  [59] &quot;NonTargeting_Human_0730&quot; &quot;NonTargeting_Human_0279&quot;
##  [61] &quot;NonTargeting_Human_0068&quot; &quot;ZNF354C&quot;                
##  [63] &quot;NonTargeting_Human_0313&quot; &quot;ZNF345&quot;                 
##  [65] &quot;NonTargeting_Human_0128&quot; &quot;NonTargeting_Human_0002&quot;
##  [67] &quot;ZNF253&quot;                  &quot;SOX14&quot;                  
##  [69] &quot;ZNF671&quot;                  &quot;NonTargeting_Human_0628&quot;
##  [71] &quot;SRY&quot;                     &quot;AIRE&quot;                   
##  [73] &quot;NonTargeting_Human_0149&quot; &quot;NonTargeting_Human_0333&quot;
##  [75] &quot;NonTargeting_Human_0904&quot; &quot;ZDHHC7&quot;                 
##  [77] &quot;NonTargeting_Human_0244&quot; &quot;NonTargeting_Human_0857&quot;
##  [79] &quot;TWIST2&quot;                  &quot;RB1&quot;                    
##  [81] &quot;NonTargeting_Human_0407&quot; &quot;ANKRD45&quot;                
##  [83] &quot;NR2F2&quot;                   &quot;NFATC3&quot;                 
##  [85] &quot;TCF12&quot;                   &quot;ZFHX2&quot;                  
##  [87] &quot;NonTargeting_Human_0914&quot; &quot;NonTargeting_Human_0467&quot;
##  [89] &quot;EGR2&quot;                    &quot;NonTargeting_Human_0153&quot;
##  [91] &quot;ZNF565&quot;                  &quot;NonTargeting_Human_0673&quot;
##  [93] &quot;MYOG&quot;                    &quot;NonTargeting_Human_0585&quot;
##  [95] &quot;NonTargeting_Human_0626&quot; &quot;NonTargeting_Human_0753&quot;
##  [97] &quot;NonTargeting_Human_0548&quot; &quot;ZNF550&quot;                 
##  [99] &quot;NonTargeting_Human_0639&quot; &quot;ZNF483&quot;                 
## [101] &quot;NonTargeting_Human_0432&quot; &quot;LHX3&quot;                   
## [103] &quot;ZNF473&quot;                  &quot;NonTargeting_Human_0537&quot;
## [105] &quot;ZBTB46&quot;                  &quot;TRIM27&quot;                 
## [107] &quot;NonTargeting_Human_0369&quot; &quot;ZNF765&quot;                 
## [109] &quot;NonTargeting_Human_0104&quot; &quot;NonTargeting_Human_0395&quot;
## [111] &quot;NonTargeting_Human_0078&quot; &quot;ZNF611&quot;                 
## [113] &quot;HMGN4&quot;                   &quot;NonTargeting_Human_0717&quot;
## [115] &quot;NonTargeting_Human_0666&quot; &quot;NonTargeting_Human_0894&quot;
## [117] &quot;HES1&quot;                    &quot;HMGB4&quot;                  
## [119] &quot;NonTargeting_Human_0501&quot; &quot;PTRF&quot;                   
## [121] &quot;NonTargeting_Human_0787&quot; &quot;NonTargeting_Human_0064&quot;
## [123] &quot;NonTargeting_Human_0387&quot; &quot;ZNF555&quot;                 
## [125] &quot;RARA&quot;                    &quot;NonTargeting_Human_0656&quot;
## [127] &quot;NonTargeting_Human_0757&quot; &quot;FBXO7&quot;                  
## [129] &quot;HIST1H1C&quot;                &quot;NonTargeting_Human_0355&quot;
## [131] &quot;SIX6&quot;                    &quot;NonTargeting_Human_0396&quot;
## [133] &quot;NonTargeting_Human_0553&quot; &quot;FOXC2&quot;                  
## [135] &quot;NonTargeting_Human_0164&quot; &quot;GATA6&quot;                  
## [137] &quot;NonTargeting_Human_0049&quot; &quot;NonTargeting_Human_0547&quot;
## [139] &quot;ZNF57&quot;                   &quot;HEY2&quot;                   
## [141] &quot;GLI4&quot;                    &quot;NonTargeting_Human_0603&quot;
## [143] &quot;NonTargeting_Human_0383&quot; &quot;NonTargeting_Human_0574&quot;
## [145] &quot;LMO4&quot;</code></pre>
<p>top 200 TFs results above, intersecting with our 120 hits.</p>
<pre class="r"><code>hits = fread(&quot;../../resources/crispr/Hitlist_20191230.csv&quot;)
intersect(tdneg.vs.iN[`neg|rank` &lt; 201, id], tdneg.vs.tdpos[`neg|rank` &lt; 201, id]) %&gt;% 
  intersect(hits$gene_name)</code></pre>
<pre><code>## [1] &quot;HSFY1&quot;  &quot;TBPL1&quot;  &quot;ZNF671&quot;</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
